{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     /users/Etu7/21200397/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package averaged_perceptron_tagger to\n",
      "[nltk_data]     /users/Etu7/21200397/nltk_data...\n",
      "[nltk_data]   Package averaged_perceptron_tagger is already up-to-\n",
      "[nltk_data]       date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     /users/Etu7/21200397/nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n",
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     /users/Etu7/21200397/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "/tmp/ipykernel_16209/1321755799.py:12: DeprecationWarning: \n",
      "Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),\n",
      "(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)\n",
      "but was not found to be installed on your system.\n",
      "If this would cause problems for you,\n",
      "please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466\n",
      "        \n",
      "  import pandas as pd\n"
     ]
    }
   ],
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.feature_extraction.text import ENGLISH_STOP_WORDS\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from nltk.tokenize import word_tokenize\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import make_scorer, f1_score, roc_auc_score, accuracy_score\n",
    "\n",
    "import numpy as np\n",
    "import utils as ut\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "\n",
    "# Ignorer tous les avertissements\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "# Ignorer les avertissements spécifiques par leur catégorie\n",
    "warnings.filterwarnings(\"ignore\", category=UserWarning)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Etape 0 : Chargement de données movies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dans cette partie nous allons rechargé les données de movies afin d'effectuer l'analyse de sentiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"../datasets/movies/movies1000/\"\n",
    "alltxts,alllabs = ut.load_movies(path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Création du DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>supposedly based on a true story in which the ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>one might expect a cathartic viewing experienc...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the camera zooms in incredibly close . \\nit fo...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1990s would remembered as the era of binary mo...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\" the world on land -- it's just too big for ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1995</th>\n",
       "      <td>one of the more unusual and suggestively viole...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1996</th>\n",
       "      <td>i find most of television so intensely boring ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1997</th>\n",
       "      <td>the event of events is upon us . \\npeople have...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1998</th>\n",
       "      <td>the sheer horrific audacity of the nazi plan t...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1999</th>\n",
       "      <td>melvin udall is a heartless man . \\nhe spends ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2000 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   text  label\n",
       "0     supposedly based on a true story in which the ...      0\n",
       "1     one might expect a cathartic viewing experienc...      0\n",
       "2     the camera zooms in incredibly close . \\nit fo...      0\n",
       "3     1990s would remembered as the era of binary mo...      0\n",
       "4      \" the world on land -- it's just too big for ...      0\n",
       "...                                                 ...    ...\n",
       "1995  one of the more unusual and suggestively viole...      1\n",
       "1996  i find most of television so intensely boring ...      1\n",
       "1997  the event of events is upon us . \\npeople have...      1\n",
       "1998  the sheer horrific audacity of the nazi plan t...      1\n",
       "1999  melvin udall is a heartless man . \\nhe spends ...      1\n",
       "\n",
       "[2000 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "movies_df = pd.DataFrame()\n",
    "movies_df['text'] = alltxts\n",
    "movies_df['label'] = alllabs\n",
    "\n",
    "movies_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Grid search pour TF-IDF Vectorizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fitting 5 folds for each of 165888 candidates, totalling 829440 fits\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "\nAll the 829440 fits failed.\nIt is very likely that your model is misconfigured.\nYou can try to debug the error by setting error_score='raise'.\n\nBelow are more details about the failures:\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'sincere', 'ten', 'can', 'former', 'even', 'again', 'along', 'seemed', 'forty', 'no', 'seeming', 'whence', 'why', 'nobody', 'about', 'part', 'none', 'towards', 'whereafter', 'never', 'empty', 'therein', 'so', 'i', 'whither', 'cannot', 'however', 'itself', 'because', 'being', 'became', 'in', 'ourselves', 'ie', 'they', 'would', 'five', 'and', 'anyone', 'thus', 'around', 'still', 'sixty', 'now', 'indeed', 'done', 'six', 'will', 'sometime', 'per', 'beyond', 'among', 'such', 'if', 'less', 'moreover', 'behind', 'here', 'un', 'etc', 'an', 'yourselves', 'has', 'we', 'found', 'wherein', 'eg', 'amoungst', 'of', 'are', 'themselves', 'another', 'off', 'under', 'where', 'have', 'enough', 'who', 'seem', 'might', 'over', 'own', 'this', 'therefore', 'often', 'system', 'eleven', 'next', 'several', 're', 'above', 'himself', 'nowhere', 'yourself', 'hereafter', 'name', 'below', 'fire', 'out', 'anyhow', 'what', 'up', 'were', 'then', 'amongst', 'put', 'hence', 'after', 'from', 'few', 'her', 'do', 'formerly', 'made', 'most', 'myself', 'during', 'into', 'within', 'must', 'throughout', 'interest', 'everywhere', 'only', 'perhaps', 'be', 'else', 'between', 'least', 'also', 'four', 'nine', 'hereby', 'should', 'via', 'side', 'same', 'whose', 'one', 'she', 'other', 'but', 'not', 'thereby', 'con', 'when', 'latter', 'get', 'all', 'for', 'twenty', 'describe', 'somewhere', 'alone', 'my', 'neither', 'amount', 'yours', 'by', 'he', 'more', 'anyway', 'others', 'any', 'him', 'find', 'everyone', 'once', 'bill', 'thin', 'detail', 'them', 'herein', 'its', 'except', 'mine', 'latterly', 'whole', 'whatever', 'hundred', 'front', 'someone', 'fill', 'had', 'keep', 'us', 'through', 'everything', 'their', 'herself', 'give', 'eight', 'toward', 'anything', 'ever', 'each', 'wherever', 'fifteen', 'somehow', 'upon', 'though', 'whereupon', 'bottom', 'although', 'de', 'me', 'third', 'yet', 'every', 'rather', 'hereupon', 'your', 'fifty', 'otherwise', 'whom', 'whereby', 'thereafter', 'whether', 'against', 'go', 'as', 'both', 'some', 'at', 'how', 'nevertheless', 'ltd', 'afterwards', 'our', 'sometimes', 'cant', 'thereupon', 'may', 'without', 'is', 'see', 'almost', 'either', 'beforehand', 'full', 'that', 'mostly', 'to', 'these', 'nothing', 'until', 'couldnt', 'becoming', 'show', 'anywhere', 'meanwhile', 'before', 'the', 'a', 'hasnt', 'or', 'was', 'those', 'last', 'whoever', 'elsewhere', 'thence', 'across', 'very', 'twelve', 'been', 'first', 'always', 'namely', 'seems', 'become', 'cry', 'two', 'move', 'there', 'with', 'since', 'serious', 'am', 'whenever', 'on', 'due', 'becomes', 'mill', 'noone', 'back', 'it', 'nor', 'too', 'three', 'which', 'please', 'while', 'onto', 'whereas', 'besides', 'thru', 'call', 'something', 'than', 'inc', 'already', 'his', 'hers', 'ours', 'co', 'thick', 'take', 'down', 'many', 'together', 'top', 'could', 'you', 'further', 'well', 'much', 'beside'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'anything', 'over', 'since', 'whom', 'neither', 'hers', 'whereas', 'anyhow', 'whereby', 'front', 'herself', 'our', 'due', 'next', 'amount', 'after', 'describe', 'hundred', 'rather', 'none', 'a', 'am', 'it', 'hence', 'nobody', 'eleven', 'along', 'mine', 'anyone', 'therefore', 'across', 'together', 'de', 'toward', 'until', 'on', 'then', 'again', 'fifteen', 'less', 'become', 'thereafter', 'your', 'onto', 'most', 'yet', 'eight', 'what', 'may', 'without', 'becomes', 'mill', 'was', 'find', 'namely', 'ever', 'up', 'seems', 'wherein', 'four', 'afterwards', 'those', 'detail', 'cry', 'thin', 'bill', 'cant', 'fire', 'whence', 'fifty', 'more', 'go', 'under', 'put', 'us', 'ten', 'themselves', 'bottom', 'besides', 'seemed', 'top', 'noone', 'even', 'my', 'do', 'much', 'empty', 'part', 'enough', 'everything', 'seem', 'only', 'moreover', 'towards', 'its', 'name', 'sometime', 'above', 'their', 'every', 'eg', 'around', 'ltd', 'though', 'through', 'this', 'beside', 'are', 'three', 'least', 'somewhere', 'wherever', 'sometimes', 'therein', 'con', 'me', 'during', 'and', 'another', 'him', 'co', 'full', 'he', 'we', 'made', 'who', 'his', 'sixty', 'whose', 'has', 'thick', 'among', 'ourselves', 'six', 'both', 'yourselves', 'either', 'once', 'although', 'see', 'somehow', 'other', 're', 'everywhere', 'all', 'from', 'hereby', 'hasnt', 'is', 'be', 'any', 'move', 'ie', 'why', 'system', 'amoungst', 'whoever', 'one', 'whole', 'anyway', 'i', 'often', 'been', 'within', 'fill', 'because', 'etc', 'take', 'yours', 'which', 'below', 'hereupon', 'latterly', 'whereafter', 'she', 'whenever', 'could', 'thereby', 'by', 'two', 'first', 'get', 'out', 'last', 'already', 'had', 'serious', 'thus', 'give', 'here', 'beforehand', 'about', 'alone', 'where', 'as', 'behind', 'keep', 'forty', 'anywhere', 'twelve', 'mostly', 'how', 'however', 'amongst', 'former', 'nevertheless', 'never', 'the', 'interest', 'so', 'down', 'but', 'sincere', 'yourself', 'also', 'into', 'show', 'further', 'herein', 'via', 'thru', 'formerly', 'seeming', 'no', 'cannot', 'will', 'with', 'there', 'against', 'these', 'couldnt', 'otherwise', 'something', 'her', 'except', 'should', 'must', 'whatever', 'of', 'have', 'an', 'each', 'many', 'such', 'found', 'almost', 'third', 'whereupon', 'beyond', 'five', 'became', 'well', 'others', 'hereafter', 'nor', 'were', 'when', 'might', 'they', 'or', 'myself', 'always', 'per', 'off', 'everyone', 'inc', 'own', 'please', 'done', 'too', 'now', 'thereupon', 'in', 'very', 'you', 'for', 'than', 'twenty', 'at', 'if', 'that', 'latter', 'someone', 'some', 'whether', 'to', 'them', 'nothing', 'himself', 'still', 'upon', 'else', 'between', 'itself', 'ours', 'elsewhere', 'call', 'throughout', 'nowhere', 'being', 'while', 'becoming', 'un', 'same', 'thence', 'indeed', 'whither', 'would', 'before', 'perhaps', 'not', 'several', 'side', 'meanwhile', 'can', 'nine', 'few', 'back'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'along', 'full', 'most', 'this', 'was', 'in', 'because', 'while', 'forty', 're', 'on', 'if', 'yourselves', 'must', 'find', 'against', 'itself', 'those', 'cant', 'by', 'thru', 'ten', 'back', 'be', 'fill', 'former', 'due', 'but', 'himself', 'elsewhere', 'three', 'ie', 'it', 'otherwise', 'everything', 'amount', 'however', 'mill', 'has', 'anyone', 'off', 'therefore', 'always', 'about', 'herself', 'wherein', 'above', 'somehow', 'why', 'detail', 'inc', 'who', 'hasnt', 'done', 'have', 'twelve', 'four', 'sometime', 'ever', 'seems', 'which', 'would', 'thereupon', 'latter', 'can', 'becoming', 'ourselves', 'take', 'towards', 'whereby', 'whereas', 'interest', 'your', 'more', 'were', 'whenever', 'nor', 'become', 'own', 'already', 'down', 'you', 'hence', 'show', 'them', 'describe', 'whereafter', 'being', 'until', 'ltd', 'beforehand', 'whole', 'his', 'me', 'first', 'all', 'their', 'something', 'myself', 'or', 'anywhere', 'via', 'namely', 'might', 'these', 'any', 'thence', 'never', 'together', 'once', 'within', 'couldnt', 'latterly', 'no', 'de', 'someone', 'are', 'only', 'amongst', 'thin', 'few', 'nine', 'among', 'every', 'around', 'fire', 'give', 'nobody', 'mostly', 'twenty', 'get', 'five', 'moreover', 'themselves', 'so', 'whereupon', 'eleven', 'per', 'enough', 'see', 'toward', 'same', 'that', 'how', 'its', 'for', 'seem', 'sincere', 'across', 'behind', 'put', 'nevertheless', 'had', 'out', 'whether', 'seeming', 'could', 'nothing', 'onto', 'top', 'indeed', 'nowhere', 'both', 'none', 'with', 'i', 'one', 'third', 'whom', 'bill', 'well', 'some', 'found', 'another', 'before', 'side', 'afterwards', 'then', 'amoungst', 'wherever', 'eight', 'a', 'con', 'such', 'everyone', 'not', 'they', 'anything', 'keep', 'without', 'hereafter', 'therein', 'noone', 'herein', 'least', 'empty', 'often', 'through', 'fifty', 'beyond', 'is', 'thereafter', 'besides', 'hers', 'whither', 'here', 'at', 'into', 'and', 'etc', 'perhaps', 'am', 'as', 'move', 'much', 'been', 'each', 'anyway', 'her', 'name', 'many', 'upon', 'below', 'everywhere', 'too', 'fifteen', 'may', 'although', 'of', 'less', 'what', 'whatever', 'to', 'whence', 'now', 'sometimes', 'should', 'neither', 'hereby', 'throughout', 'my', 'between', 'when', 'anyhow', 'very', 'cannot', 'thick', 'do', 'hereupon', 'two', 'us', 'last', 'next', 'we', 'formerly', 'serious', 'meanwhile', 'go', 'under', 'than', 'yet', 'up', 'system', 'over', 'will', 'after', 'except', 'whose', 'sixty', 'further', 'from', 'she', 'six', 'thereby', 'during', 'alone', 'somewhere', 'several', 'beside', 'since', 'yourself', 'whoever', 'front', 'co', 'seemed', 'an', 'the', 'rather', 'made', 'even', 'hundred', 'bottom', 'mine', 'almost', 'eg', 'again', 'else', 'our', 'un', 'also', 'part', 'where', 'please', 'others', 'cry', 'either', 'still', 'thus', 'call', 'though', 'becomes', 'him', 'ours', 'there', 'yours', 'he', 'other', 'became'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'done', 'hereby', 'eg', 'front', 'found', 'someone', 'amongst', 'nor', 'until', 'ltd', 'within', 'among', 'so', 'other', 'do', 'either', 'can', 'cannot', 'amount', 'thereupon', 'whenever', 'during', 'herein', 'twelve', 'many', 'onto', 'may', 'whole', 'my', 'them', 'were', 'its', 'four', 'therefore', 'what', 'latterly', 'with', 'hers', 'go', 'if', 'beyond', 'not', 'nevertheless', 'themselves', 'this', 'thence', 'co', 'one', 'once', 'none', 'why', 'above', 'could', 'however', 'mostly', 'very', 'although', 'formerly', 'thereby', 'ourselves', 'hundred', 'when', 'serious', 'beside', 'who', 'myself', 'such', 'your', 'becomes', 'thus', 'therein', 'first', 'nowhere', 'off', 'yours', 'take', 'how', 'next', 'had', 'me', 'five', 'that', 'whereupon', 'call', 'seem', 'ever', 'amoungst', 'i', 'couldnt', 'some', 'than', 'further', 'all', 'any', 'only', 'detail', 'the', 'sixty', 'describe', 'full', 'inc', 'whereafter', 'well', 'indeed', 'becoming', 'perhaps', 'meanwhile', 'everyone', 'am', 'please', 'most', 'de', 'etc', 'which', 'thru', 'throughout', 'no', 'she', 'on', 'yourselves', 'against', 'seemed', 'seems', 'towards', 'another', 'already', 'whence', 'cry', 'it', 'been', 'own', 'have', 'name', 'elsewhere', 'whatever', 'out', 'mill', 'get', 'thick', 'mine', 'often', 'sincere', 'thereafter', 'yet', 'twenty', 'enough', 'in', 'ours', 'third', 'except', 'part', 'her', 'up', 'afterwards', 'neither', 'under', 'latter', 'give', 'again', 'was', 'every', 'less', 'put', 'being', 'noone', 'back', 'then', 'fifty', 'hereupon', 'sometime', 'we', 'anyway', 'into', 'around', 'together', 'they', 'fill', 'everything', 'namely', 'whether', 'interest', 'behind', 'also', 'whereas', 'beforehand', 'through', 'wherein', 'always', 'almost', 'even', 'because', 'forty', 'whose', 'ie', 'us', 'per', 'side', 'last', 'fire', 'before', 'will', 'see', 'moreover', 'be', 'else', 'bottom', 'below', 'become', 'as', 'more', 'fifteen', 'both', 'him', 'though', 'hasnt', 'others', 'due', 'whereby', 'somewhere', 're', 'here', 'least', 'nobody', 'con', 'few', 'to', 'over', 'system', 'of', 'three', 'still', 'whoever', 'became', 'must', 'besides', 'you', 'top', 'itself', 'made', 'while', 'two', 'former', 'bill', 'toward', 'eleven', 'too', 'between', 'much', 'an', 'for', 'something', 'and', 'from', 'cant', 'eight', 'empty', 'our', 'move', 'about', 'there', 'his', 'are', 'thin', 'hereafter', 'whither', 'sometimes', 'otherwise', 'un', 'along', 'seeming', 'alone', 'same', 'down', 'anyone', 'yourself', 'keep', 'find', 'upon', 'these', 'without', 'would', 'across', 'anything', 'or', 'by', 'anyhow', 'several', 'is', 'wherever', 'where', 'those', 'herself', 'hence', 'somehow', 'each', 'but', 'after', 'ten', 'he', 'should', 'everywhere', 'himself', 'since', 'via', 'has', 'might', 'never', 'a', 'now', 'nine', 'whom', 'six', 'rather', 'show', 'at', 'nothing', 'their', 'anywhere'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'when', 'indeed', 'something', 'here', 'someone', 'do', 'eleven', 'become', 'across', 'i', 'too', 'except', 'such', 'up', 'himself', 'show', 'several', 'done', 'empty', 'in', 'although', 'anyhow', 'whence', 'after', 'there', 'behind', 'other', 'under', 'an', 'ie', 'see', 'hers', 'describe', 'ours', 'by', 'within', 'few', 'hereafter', 'less', 'detail', 'any', 'top', 'each', 'without', 'where', 'can', 'hence', 'me', 'his', 'herein', 'serious', 'un', 'meanwhile', 'many', 'sometimes', 'besides', 'you', 'amongst', 'yourself', 'last', 'along', 'why', 'enough', 'around', 'part', 'whether', 'then', 'for', 'against', 'together', 'everything', 'been', 'are', 'one', 'find', 'everyone', 'more', 'their', 'be', 'whatever', 'same', 'keep', 'else', 'until', 'latterly', 'thereafter', 'bill', 'nor', 'she', 'thin', 'nobody', 'back', 'could', 'due', 'yours', 'everywhere', 'your', 'ltd', 'sixty', 'to', 'almost', 'between', 'three', 'full', 'ourselves', 'while', 'should', 'as', 'via', 'anything', 'must', 'has', 'him', 'because', 'moreover', 'itself', 'now', 'six', 'con', 'whereupon', 'amount', 'fire', 'nothing', 'give', 'thus', 'inc', 'eight', 'nine', 'becomes', 'former', 'how', 'forty', 'among', 'is', 'very', 'side', 're', 'found', 'might', 'if', 'afterwards', 'he', 'seemed', 'of', 'thence', 'who', 'cry', 'either', 'beside', 'go', 'was', 'perhaps', 'anyway', 'would', 'co', 'through', 'with', 'so', 'towards', 'next', 'however', 'sometime', 'twelve', 'name', 'themselves', 'seems', 'eg', 'a', 'down', 'than', 'thick', 'anyone', 'every', 'never', 'but', 'therein', 'those', 'cant', 'others', 'since', 'much', 'wherever', 'seem', 'neither', 'all', 'further', 'above', 'get', 'first', 'made', 'take', 'formerly', 'interest', 'we', 'even', 'once', 'thru', 'below', 'whenever', 'whom', 'de', 'nevertheless', 'least', 'twenty', 'will', 'third', 'move', 'always', 'otherwise', 'beyond', 'fifteen', 'five', 'only', 'elsewhere', 'had', 'another', 'put', 'what', 'hasnt', 'alone', 'became', 'anywhere', 'mill', 'own', 'upon', 'throughout', 'already', 'per', 'ever', 'well', 'most', 'off', 'whose', 'yourselves', 'am', 'none', 'whereafter', 'they', 'again', 'herself', 'still', 'this', 'have', 'therefore', 'my', 'thereby', 'toward', 'our', 'namely', 'becoming', 'at', 'wherein', 'us', 'mine', 'it', 'couldnt', 'ten', 'whoever', 'though', 'seeming', 'two', 'etc', 'rather', 'noone', 'nowhere', 'hereby', 'whereas', 'bottom', 'also', 'which', 'often', 'please', 'latter', 'being', 'fill', 'somewhere', 'over', 'about', 'system', 'some', 'cannot', 'fifty', 'myself', 'thereupon', 'not', 'sincere', 'beforehand', 'onto', 'that', 'somehow', 'no', 'on', 'hereupon', 'them', 'may', 'amoungst', 'into', 'her', 'whole', 'its', 'mostly', 'during', 'these', 'whereby', 'call', 'both', 'out', 'were', 'and', 'before', 'front', 'four', 'yet', 'or', 'whither', 'from', 'hundred', 'the'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'is', 'ie', 'your', 'should', 'since', 'call', 'be', 'eg', 'when', 'most', 'us', 'two', 'against', 'own', 'whither', 'after', 'enough', 'first', 'elsewhere', 'about', 'other', 'detail', 'together', 'put', 'who', 'thin', 'always', 'either', 'on', 'rather', 'you', 'meanwhile', 'whence', 'thence', 'except', 'because', 'whereupon', 'between', 'seeming', 'whoever', 'cannot', 'last', 'mill', 'fire', 'herself', 'whereas', 'few', 'through', 'became', 'along', 'with', 'we', 'moreover', 'being', 'hereafter', 'therein', 'or', 'empty', 'please', 'whose', 'too', 'not', 'had', 'could', 'hence', 'among', 'around', 'amongst', 'and', 'twelve', 'themselves', 'as', 'sincere', 'he', 'of', 'take', 'whenever', 'my', 'never', 'neither', 'get', 'its', 'go', 'have', 'couldnt', 'one', 'would', 'full', 'before', 'whole', 'am', 'seems', 'no', 'across', 'where', 'upon', 'down', 'but', 'yourself', 'something', 'an', 'bottom', 'each', 'although', 'thereafter', 'wherein', 'for', 'fifty', 'latter', 'inc', 'therefore', 'etc', 'whatever', 'ours', 'else', 'hers', 'three', 'sometimes', 'they', 'formerly', 'former', 'it', 'whom', 'up', 'becomes', 'myself', 'however', 'alone', 'this', 'system', 'over', 'find', 'per', 'his', 'interest', 'the', 'amount', 'though', 'due', 'found', 'him', 'forty', 'con', 'off', 'wherever', 'anyone', 'both', 'hundred', 'give', 'besides', 'yourselves', 'more', 'another', 'everyone', 'toward', 'perhaps', 'many', 'here', 'nobody', 'were', 'some', 'again', 'hereupon', 'can', 'name', 'even', 'becoming', 'must', 'throughout', 'fill', 'has', 'ever', 'everywhere', 'nevertheless', 'indeed', 'itself', 'if', 'only', 'ten', 'via', 'whereby', 'mostly', 'further', 'whereafter', 'until', 're', 'why', 'amoungst', 'serious', 'also', 'others', 'well', 'anyhow', 'will', 'that', 'de', 'above', 'mine', 'during', 'into', 'beside', 'yours', 'nor', 'i', 'nine', 'see', 'four', 'hasnt', 'out', 'whether', 'while', 'any', 'already', 'do', 'ourselves', 'are', 'how', 'namely', 'very', 'show', 'eleven', 'latterly', 'thereupon', 'them', 'same', 'which', 'seem', 'beforehand', 'thru', 'move', 'somehow', 'next', 'within', 'keep', 'least', 'third', 'below', 'describe', 'almost', 'bill', 'beyond', 'those', 'eight', 'all', 'onto', 'yet', 'then', 'noone', 'sixty', 'by', 'part', 'now', 'still', 'done', 'top', 'cant', 'himself', 'towards', 'everything', 'back', 'she', 'somewhere', 'ltd', 'than', 'in', 'what', 'me', 'might', 'six', 'thick', 'anywhere', 'twenty', 'none', 'such', 'often', 'thereby', 'anything', 'co', 'so', 'nowhere', 'herein', 'un', 'every', 'there', 'sometime', 'thus', 'under', 'a', 'hereby', 'otherwise', 'less', 'seemed', 'our', 'may', 'made', 'five', 'these', 'nothing', 'her', 'side', 'several', 'cry', 'once', 'become', 'their', 'front', 'behind', 'at', 'from', 'without', 'anyway', 'someone', 'afterwards', 'to', 'was', 'much', 'been', 'fifteen'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'should', 'each', 'eight', 'four', 'wherein', 'cant', 'are', 'to', 'describe', 'whole', 'thereupon', 'keep', 'been', 'indeed', 'moreover', 'ever', 'mill', 'by', 'with', 'go', 'de', 'become', 'anyone', 'latterly', 'these', 'though', 'hence', 'thin', 'among', 'wherever', 'some', 'because', 'every', 'the', 'either', 'therefore', 'whereas', 'un', 'done', 'beyond', 'before', 'sincere', 'otherwise', 'noone', 'thereafter', 'i', 'onto', 'twenty', 'was', 'himself', 'would', 'often', 'mostly', 'inc', 'ours', 'nor', 'must', 'has', 'nine', 'latter', 'me', 'amount', 'yours', 'below', 'find', 'myself', 'few', 'over', 'former', 'both', 'afterwards', 'someone', 'under', 'found', 'ltd', 'itself', 'bottom', 'of', 'please', 'fill', 'for', 'without', 'thence', 'as', 'my', 'cry', 'something', 'yet', 'whenever', 'into', 'anyway', 'here', 'mine', 'am', 'after', 'nobody', 'everyone', 'meanwhile', 'eg', 'within', 'very', 'system', 'what', 'five', 'fifty', 'therein', 'or', 'their', 'one', 'whereafter', 'her', 'up', 'name', 'always', 'even', 'yourself', 'nothing', 'have', 'across', 'eleven', 'others', 'made', 'etc', 'last', 'becoming', 'more', 'can', 'too', 'amoungst', 'alone', 'then', 'empty', 'in', 'against', 'whatever', 'throughout', 'sixty', 'us', 'everything', 'six', 'namely', 'had', 'his', 'side', 'since', 'own', 'except', 'sometimes', 'seeming', 'out', 'and', 'via', 'might', 'how', 'our', 'do', 'move', 'together', 'where', 'is', 'anything', 'thereby', 'further', 'those', 'perhaps', 'seems', 'another', 'they', 'he', 'many', 'see', 'least', 'who', 'through', 'due', 'while', 'cannot', 'however', 'anyhow', 'almost', 'sometime', 'somehow', 'formerly', 'whoever', 'fire', 'a', 'still', 'amongst', 'between', 'towards', 'whereupon', 'much', 'your', 'which', 'else', 'thus', 'forty', 'hereupon', 'thru', 'whereby', 'besides', 'them', 'most', 'you', 'above', 'at', 'couldnt', 'none', 'themselves', 'its', 'top', 'along', 'again', 'off', 'part', 'there', 'about', 'once', 'show', 'she', 'if', 'never', 'detail', 'when', 'seemed', 'any', 'upon', 'back', 'it', 'rather', 're', 'full', 'two', 'hereby', 'hers', 'whether', 'herein', 'than', 'behind', 'so', 'well', 'an', 'now', 'somewhere', 'why', 'became', 'next', 'elsewhere', 'neither', 'yourselves', 'call', 'him', 'also', 'from', 'twelve', 'we', 'being', 'ten', 'be', 'other', 'no', 'take', 'whence', 'anywhere', 'three', 'nevertheless', 'get', 'hundred', 'ie', 'bill', 'fifteen', 'becomes', 'whom', 'this', 'co', 'several', 'third', 'were', 'per', 'whose', 'beside', 'during', 'less', 'around', 'will', 'only', 'serious', 'on', 'put', 'interest', 'thick', 'whither', 'nowhere', 'down', 'could', 'hasnt', 'same', 'until', 'not', 'give', 'that', 'con', 'such', 'all', 'although', 'herself', 'everywhere', 'ourselves', 'first', 'but', 'toward', 'may', 'seem', 'enough', 'already', 'front', 'hereafter', 'beforehand'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'four', 'de', 'seems', 'they', 'out', 'fifty', 'be', 'first', 'empty', 'are', 'became', 'himself', 'describe', 'although', 'nobody', 'something', 'him', 'been', 'were', 'that', 'thereupon', 'only', 'meanwhile', 'formerly', 'moreover', 'yourself', 'but', 'well', 'the', 'perhaps', 'somewhere', 'forty', 'inc', 'into', 'more', 'cant', 'hence', 'than', 'have', 'everything', 'in', 'them', 're', 'serious', 'one', 'it', 'thereby', 'anything', 'ltd', 'keep', 'ie', 'eight', 'top', 'otherwise', 'if', 'call', 'where', 'hers', 'i', 'eg', 'though', 'find', 'yourselves', 'next', 'may', 'alone', 'third', 'your', 'indeed', 'whereupon', 'cry', 'to', 'six', 'should', 'hereby', 'etc', 'whole', 'whose', 'off', 'as', 'hereafter', 'side', 'and', 'beforehand', 'give', 'now', 'become', 'least', 'co', 'until', 'full', 'together', 'others', 'mine', 'am', 'which', 'themselves', 'myself', 'sometime', 'whereby', 'do', 'nine', 'sincere', 'ever', 'latterly', 'wherever', 'no', 'last', 'whenever', 'seeming', 'for', 'an', 'this', 'mill', 'every', 'within', 'many', 'has', 'down', 'onto', 'ten', 'namely', 'seem', 'on', 'everywhere', 'however', 'who', 'go', 'anyhow', 'fifteen', 'whatever', 'whereas', 'twelve', 'via', 'beside', 'mostly', 'bill', 'amount', 'towards', 'us', 'besides', 'both', 'during', 'whither', 'most', 'upon', 'not', 'rather', 'how', 'since', 'what', 'enough', 'or', 'across', 'often', 'system', 'take', 'several', 'couldnt', 'show', 'too', 'hereupon', 'throughout', 'nowhere', 'by', 'around', 'three', 'interest', 'seemed', 'would', 'our', 'found', 'per', 'his', 'hasnt', 'from', 'wherein', 'somehow', 'at', 'between', 'never', 'becoming', 'was', 'still', 'her', 'someone', 'under', 'even', 'before', 'had', 'thick', 'through', 'about', 'again', 'such', 'two', 'noone', 'none', 'ourselves', 'almost', 'nor', 'yours', 'she', 'afterwards', 'own', 'anywhere', 'anyone', 'toward', 'why', 'against', 'amongst', 'move', 'very', 'along', 'whence', 'is', 'being', 'any', 'already', 'hundred', 'up', 'there', 'its', 'among', 'other', 'much', 'please', 'might', 'my', 'everyone', 'amoungst', 'yet', 'another', 'he', 'nothing', 'those', 'over', 'see', 'then', 'when', 'their', 'beyond', 'these', 'always', 'less', 'also', 'so', 'behind', 'cannot', 'will', 'thin', 'sometimes', 'therefore', 'un', 'thus', 'due', 'name', 'fire', 'whereafter', 'con', 'else', 'detail', 'below', 'sixty', 'few', 'latter', 'eleven', 'you', 'ours', 'five', 'back', 'we', 'herein', 'herself', 'while', 'further', 'without', 'all', 'of', 'after', 'because', 'twenty', 'except', 'done', 'here', 'whether', 'must', 'elsewhere', 'once', 'either', 'some', 'with', 'can', 'made', 'itself', 'above', 'fill', 'neither', 'whom', 'could', 'former', 'put', 'therein', 'thereafter', 'becomes', 'get', 'anyway', 'same', 'thru', 'me', 'part', 'each', 'nevertheless', 'thence', 'front', 'a', 'bottom', 'whoever'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'cant', 'much', 'every', 'first', 'several', 'had', 'how', 'hers', 'side', 'around', 'eg', 'keep', 'onto', 'fifteen', 'either', 'few', 'mine', 'here', 'now', 'during', 'were', 'etc', 'enough', 'thereafter', 'you', 'these', 'between', 'ever', 'whereby', 'me', 'while', 'beside', 'she', 'hereupon', 'everyone', 'must', 'indeed', 'most', 'behind', 'give', 'four', 'very', 'though', 'before', 'alone', 'with', 'bill', 'been', 'out', 'although', 'detail', 'being', 'too', 'becomes', 'this', 'only', 'his', 'where', 'hereby', 'describe', 'by', 'do', 'it', 'myself', 'eleven', 'nothing', 'fill', 'more', 'whereupon', 'perhaps', 'each', 'another', 'would', 'to', 'since', 'all', 'whatever', 'ltd', 'the', 'always', 'towards', 'never', 'and', 'thus', 're', 'co', 'whenever', 'eight', 'somewhere', 'then', 'should', 'everything', 'something', 'whither', 'we', 'have', 'without', 'who', 'hereafter', 'was', 'no', 'also', 'may', 'because', 'take', 'nevertheless', 'meanwhile', 'twelve', 'down', 'against', 'done', 'when', 'he', 'herein', 'through', 'whom', 'anyhow', 'get', 'twenty', 'into', 'latterly', 'fifty', 'yours', 'bottom', 'my', 'full', 'interest', 'still', 'elsewhere', 'found', 'could', 'within', 'hasnt', 'nor', 'whoever', 'might', 'amoungst', 'seems', 'forty', 'due', 'until', 'among', 'six', 'for', 'about', 'moreover', 'next', 'over', 'which', 'inc', 'rather', 'per', 'less', 'throughout', 'below', 'a', 'almost', 'others', 'back', 'her', 'see', 'however', 'else', 'why', 'name', 'nine', 'move', 'is', 'us', 'neither', 'cannot', 'un', 'yourselves', 'third', 'seemed', 'along', 'thick', 'whose', 'whether', 'namely', 'sometime', 'has', 'what', 'mill', 'already', 'be', 'are', 'any', 'fire', 'at', 'itself', 'even', 'if', 'but', 'thereby', 'such', 'off', 'once', 'toward', 'via', 'one', 'so', 'wherein', 'an', 'except', 'serious', 'beyond', 'sincere', 'afterwards', 'ten', 'go', 'top', 'hundred', 'nobody', 'wherever', 'him', 'am', 'front', 'latter', 'anywhere', 'across', 'will', 'part', 'none', 'thru', 'whence', 'some', 'yourself', 'or', 'yet', 'somehow', 'again', 'empty', 'under', 'became', 'they', 'thence', 'de', 'thin', 'call', 'system', 'them', 'own', 'upon', 'amongst', 'its', 'anyone', 'many', 'as', 'otherwise', 'cry', 'that', 'other', 'former', 'therefore', 'everywhere', 'there', 'herself', 'put', 'those', 'our', 'than', 'two', 'show', 'after', 'of', 'anyway', 'three', 'made', 'often', 'formerly', 'whereafter', 'thereupon', 'i', 'ie', 'seem', 'noone', 'themselves', 'becoming', 'please', 'your', 'their', 'from', 'anything', 'find', 'ours', 'besides', 'amount', 'whole', 'on', 'least', 'five', 'both', 'whereas', 'same', 'ourselves', 'hence', 'in', 'further', 'seeming', 'someone', 'mostly', 'beforehand', 'well', 'couldnt', 'together', 'up', 'can', 'sometimes', 'con', 'sixty', 'last', 'himself', 'become', 'not', 'therein', 'nowhere', 'above'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'along', 'his', 'same', 'whoever', 'anyone', 'former', 'anything', 'whom', 'everything', 'put', 'via', 'among', 'if', 'these', 'ours', 'will', 'an', 'due', 'into', 'well', 'herein', 'we', 'five', 'am', 'nevertheless', 'anyway', 'formerly', 'cant', 'how', 'such', 'between', 'my', 'amount', 'nowhere', 'very', 'is', 'after', 'side', 'whereafter', 'always', 'themselves', 'neither', 'bill', 'should', 'a', 'by', 'there', 'hereupon', 'hundred', 'the', 'all', 'except', 'whole', 'besides', 'many', 'whose', 'few', 'has', 'us', 'every', 'seemed', 'often', 'of', 'your', 'sixty', 'anyhow', 'who', 'amoungst', 'please', 'are', 'when', 'upon', 'ten', 'must', 'somewhere', 'thereafter', 'whether', 'sometime', 'however', 'de', 'nobody', 'both', 'three', 'out', 'find', 'before', 'fill', 'un', 'becomes', 'whatever', 'whence', 'four', 'whither', 'most', 'eg', 'but', 're', 'part', 'or', 'throughout', 'nothing', 'get', 'hereafter', 'something', 'latter', 'namely', 'beside', 'co', 'also', 'other', 'inc', 'below', 'found', 'meanwhile', 'serious', 'give', 'so', 'third', 'cry', 'keep', 'can', 'more', 'our', 'about', 'afterwards', 'him', 'whereas', 'been', 'may', 'not', 'go', 'whereupon', 'twelve', 'whereby', 'con', 'off', 'in', 'moreover', 'further', 'seems', 'be', 'nor', 'fire', 'detail', 'and', 'their', 'onto', 'back', 'last', 'see', 'done', 'because', 'otherwise', 'whenever', 'wherever', 'he', 'me', 'through', 'move', 'have', 'only', 'no', 'hers', 'which', 'describe', 'etc', 'too', 'indeed', 'own', 'rather', 'it', 'thru', 'she', 'to', 'even', 'everywhere', 'sometimes', 'down', 'as', 'this', 'across', 'myself', 'somehow', 'ltd', 'you', 'bottom', 'becoming', 'being', 'front', 'under', 'none', 'yourself', 'eight', 'above', 'amongst', 'her', 'behind', 'forty', 'others', 'seeming', 'either', 'fifteen', 'again', 'some', 'per', 'they', 'therefore', 'couldnt', 'nine', 'why', 'someone', 'else', 'show', 'them', 'cannot', 'twenty', 'here', 'might', 'then', 'interest', 'another', 'while', 'with', 'those', 'beyond', 'that', 'its', 'from', 'beforehand', 'each', 'up', 'herself', 'thence', 'six', 'though', 'noone', 'was', 'latterly', 'never', 'one', 'itself', 'i', 'elsewhere', 'take', 'eleven', 'first', 'for', 'next', 'within', 'everyone', 'thereupon', 'mostly', 'where', 'system', 'already', 'could', 'therein', 'together', 'two', 'than', 'wherein', 'would', 'fifty', 'hence', 'any', 'seem', 'yours', 'mine', 'himself', 'become', 'ourselves', 'until', 'do', 'alone', 'over', 'almost', 'now', 'full', 'still', 'at', 'during', 'had', 'mill', 'much', 'least', 'thick', 'toward', 'thereby', 'once', 'without', 'were', 'ever', 'yet', 'perhaps', 'on', 'anywhere', 'made', 'since', 'sincere', 'ie', 'less', 'towards', 'hasnt', 'although', 'several', 'name', 'thin', 'empty', 'enough', 'top', 'hereby', 'call', 'thus', 'yourselves', 'became', 'around', 'against', 'what'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'were', 'in', 'sometime', 'therefore', 'be', 'his', 'less', 'thereupon', 'otherwise', 'might', 'fire', 'yet', 'top', 'somehow', 'next', 'at', 'all', 'i', 'put', 'too', 'by', 'bottom', 'beforehand', 'hasnt', 'could', 'please', 'nevertheless', 'became', 'throughout', 'mine', 'even', 'fifteen', 'wherein', 'should', 'would', 'anyone', 'made', 'me', 'behind', 'thence', 'thereafter', 'along', 'ours', 'may', 'hers', 'formerly', 'yourselves', 'thick', 'not', 'whom', 'something', 'four', 'yourself', 'than', 'thus', 'among', 'eg', 'below', 'neither', 'with', 'every', 'others', 'myself', 'already', 'least', 'you', 'has', 'him', 'off', 'back', 'move', 'name', 'almost', 'again', 'and', 'therein', 'seeming', 'moreover', 'everything', 'is', 'such', 'enough', 'other', 'forty', 'perhaps', 'often', 'get', 'herself', 'my', 'some', 'noone', 'done', 'us', 'sometimes', 'do', 'the', 'everywhere', 'seemed', 'themselves', 'whereas', 'after', 'then', 'etc', 'amount', 'six', 'either', 'somewhere', 'except', 'twelve', 'onto', 'against', 're', 'are', 'same', 'once', 'serious', 'amongst', 'through', 'fill', 'third', 'sixty', 'ie', 'so', 'upon', 'beyond', 'hence', 'due', 'nor', 'go', 'was', 'becomes', 'mostly', 'which', 'amoungst', 'being', 'if', 'becoming', 'a', 'further', 'ourselves', 'must', 'several', 'anyway', 'well', 'as', 'that', 'also', 'why', 'yours', 'nothing', 'couldnt', 'into', 'someone', 'detail', 'whatever', 'give', 'your', 'meanwhile', 'since', 'five', 'will', 'she', 'but', 'together', 'eight', 'now', 'no', 'few', 'beside', 'over', 'toward', 'what', 'he', 'here', 'besides', 'during', 'itself', 'whereupon', 'ten', 'out', 'another', 'very', 'become', 'afterwards', 'alone', 'everyone', 'still', 'cant', 'rather', 'on', 'inc', 'side', 'this', 'twenty', 'part', 'three', 'up', 'none', 'himself', 'ever', 'can', 'more', 'show', 'interest', 'across', 'around', 'been', 'whose', 'towards', 'anything', 'without', 'mill', 'whither', 'for', 'see', 'nobody', 'who', 'we', 'from', 'call', 'where', 'nine', 'when', 'of', 'former', 'last', 'herein', 'it', 'to', 'hereby', 'else', 'anywhere', 'keep', 'thin', 'whence', 'before', 'take', 'empty', 'one', 'down', 'front', 'only', 'any', 'eleven', 'they', 'thru', 'ltd', 'indeed', 'many', 'its', 'elsewhere', 'above', 'hundred', 'anyhow', 'via', 'our', 'these', 'bill', 'system', 'about', 'find', 'nowhere', 'wherever', 'much', 'cry', 'hereafter', 'never', 'per', 'co', 'con', 'though', 'their', 'namely', 'thereby', 'an', 'while', 'under', 'latterly', 'how', 'most', 'them', 'seems', 'had', 'between', 'however', 'whoever', 'whereby', 'de', 'those', 'whereafter', 'am', 'describe', 'whether', 'whenever', 'first', 'cannot', 'or', 'found', 'seem', 'own', 'although', 'fifty', 'whole', 'both', 'each', 'sincere', 'always', 'latter', 'un', 'there', 'because', 'until', 'her', 'full', 'two', 'hereupon', 'have', 'within'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'found', 'ours', 'everything', 'alone', 'around', 'perhaps', 'same', 'ever', 'nor', 'else', 'latterly', 'seemed', 'all', 'eg', 'either', 'whole', 'side', 'her', 'whereafter', 'up', 'everywhere', 'some', 'was', 'along', 'as', 'more', 'therefore', 'fifteen', 'thereafter', 'otherwise', 'former', 'each', 'latter', 'sixty', 'two', 'thick', 'other', 'whom', 'nothing', 'never', 'top', 'why', 'has', 'go', 'me', 'his', 'move', 'too', 'mill', 'into', 'next', 'others', 'had', 'were', 'became', 'no', 'above', 'you', 'moreover', 'twelve', 'both', 'thereupon', 'further', 'ten', 'before', 'since', 'whoever', 'very', 'therein', 'itself', 'thence', 'about', 'could', 'so', 'nevertheless', 'un', 'behind', 'thereby', 'interest', 'in', 'becomes', 'a', 'beforehand', 'much', 'forty', 'enough', 'between', 'everyone', 'also', 'they', 'might', 'none', 'one', 'them', 'wherein', 'thus', 'full', 'hereby', 'us', 'yourselves', 'con', 'part', 'hereafter', 'become', 'couldnt', 'there', 'somewhere', 'their', 'how', 'eight', 'if', 'across', 'when', 'bottom', 'formerly', 'fill', 'hasnt', 'few', 'towards', 'via', 'themselves', 'off', 'bill', 'always', 'until', 'although', 'three', 'last', 'seeming', 'something', 'hence', 'made', 'besides', 'herself', 'yet', 'least', 'ourselves', 'de', 'namely', 'however', 'under', 'anyway', 'most', 'show', 'whither', 'its', 'even', 'am', 'that', 'another', 'from', 'must', 'my', 'toward', 'but', 'beyond', 'put', 'hers', 'here', 'she', 'over', 'below', 'co', 'ltd', 'though', 'during', 'again', 'once', 'on', 'amongst', 'per', 'thru', 'these', 'sometime', 'third', 'amoungst', 'within', 'together', 'seem', 'i', 'ie', 'seems', 'the', 'whose', 'yourself', 'anywhere', 'upon', 'already', 'whence', 'have', 'hereupon', 'name', 'five', 'call', 'detail', 'is', 'throughout', 'because', 'somehow', 'well', 'give', 'describe', 'he', 'every', 'down', 'it', 'still', 'inc', 'whereby', 'mostly', 'anyone', 'whatever', 'beside', 'may', 'front', 'after', 'who', 'four', 'eleven', 'nowhere', 'many', 'etc', 'wherever', 'get', 'and', 'afterwards', 'himself', 'to', 'only', 'less', 'what', 'cry', 'thin', 'for', 'now', 'whether', 'against', 'at', 'him', 'system', 'out', 'do', 'back', 'almost', 'herein', 'indeed', 'not', 'due', 'this', 'someone', 'nine', 'being', 'neither', 'please', 'by', 'fire', 'elsewhere', 'your', 'rather', 'myself', 'nobody', 'find', 'than', 'where', 'take', 'we', 'are', 'empty', 'six', 'will', 'whereupon', 'should', 'cannot', 'anyhow', 'first', 'except', 'twenty', 'serious', 'sincere', 'done', 'which', 'with', 'several', 'keep', 'onto', 'been', 'cant', 'without', 'mine', 'be', 'whereas', 'noone', 'through', 'yours', 'our', 'hundred', 'amount', 'among', 'own', 'can', 'becoming', 'would', 'anything', 'of', 'an', 're', 'meanwhile', 'such', 'see', 'then', 'whenever', 'often', 'while', 'fifty', 'sometimes', 'or', 'those', 'any'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'thick', 'my', 'become', 'same', 'below', 'about', 'sometimes', 'describe', 'amount', 'made', 'own', 'less', 'others', 'behind', 'sixty', 'etc', 'nowhere', 'a', 'within', 'yours', 'ltd', 'so', 'keep', 'ours', 'they', 'becoming', 'their', 'whom', 'his', 'whereafter', 'the', 'too', 'anywhere', 'during', 'either', 'formerly', 'but', 'what', 'ie', 'she', 'anyway', 'one', 'himself', 'against', 'four', 'move', 'i', 'was', 'seems', 'could', 'cannot', 'otherwise', 'find', 'whereupon', 'get', 'somewhere', 'again', 'most', 'whereas', 'mostly', 'un', 'are', 'already', 'thence', 'since', 'neither', 'herself', 'nor', 'two', 'them', 'together', 'wherever', 'alone', 'everywhere', 'than', 'whatever', 'her', 'be', 'per', 'inc', 'someone', 'some', 'thus', 'therefore', 'call', 'whenever', 'eight', 'namely', 'also', 'if', 'front', 'with', 'over', 'whence', 'once', 'whoever', 'along', 'hereupon', 'next', 'though', 'elsewhere', 'may', 'nothing', 'de', 'toward', 'else', 'twenty', 'hereafter', 'yourselves', 'around', 'those', 'although', 're', 'whether', 'latter', 'empty', 'co', 'would', 'besides', 'twelve', 'well', 'of', 'detail', 'without', 'out', 'mine', 'under', 'fill', 'why', 'put', 'always', 'we', 'where', 'he', 'never', 'full', 'hereby', 'something', 'more', 'indeed', 'while', 'that', 'myself', 'eg', 'name', 'somehow', 'found', 'latterly', 'amoungst', 'cry', 'will', 'back', 'several', 'might', 'must', 'thru', 'has', 'three', 'towards', 'amongst', 'or', 'couldnt', 'see', 'am', 'many', 'third', 'further', 'before', 'top', 'much', 'interest', 'hence', 'hers', 'among', 'him', 'should', 'seeming', 'then', 'few', 'another', 'very', 'whose', 'as', 'do', 'via', 'this', 'none', 'at', 'eleven', 'across', 'until', 'nobody', 'above', 'yet', 'six', 'thereupon', 'last', 'nevertheless', 'least', 'whereby', 'these', 'any', 'afterwards', 'up', 'it', 'except', 'and', 'hundred', 'often', 'anyone', 'an', 'mill', 'ourselves', 'former', 'when', 'side', 'seemed', 'sincere', 'show', 'onto', 'whither', 'to', 'not', 'by', 'seem', 'after', 'itself', 'beside', 'fifty', 'throughout', 'thin', 'everyone', 'have', 'yourself', 'noone', 'you', 'how', 'even', 'only', 'its', 'done', 'each', 'however', 'here', 'us', 'now', 'take', 'go', 'becomes', 'down', 'which', 'fire', 'is', 'on', 'almost', 'anything', 'hasnt', 'there', 'in', 'fifteen', 'give', 'who', 'both', 'first', 'sometime', 'themselves', 'wherein', 'con', 'because', 'bottom', 'no', 'anyhow', 'being', 'nine', 'due', 'ten', 'into', 'part', 'such', 'me', 'forty', 'our', 'perhaps', 'every', 'upon', 'all', 'ever', 'been', 'herein', 'therein', 'can', 'beyond', 'became', 'everything', 'bill', 'please', 'serious', 'beforehand', 'from', 'system', 'through', 'meanwhile', 'were', 'your', 'thereafter', 'enough', 'other', 'still', 'had', 'whole', 'thereby', 'between', 'cant', 'off', 'five', 'rather', 'for', 'moreover'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'via', 'already', 'anyone', 'above', 'most', 'may', 'nobody', 'onto', 'yourself', 'none', 'fifteen', 'nowhere', 'call', 'now', 'yourselves', 'by', 'fifty', 'here', 'each', 'herself', 'which', 'get', 'anyhow', 'someone', 'anything', 'side', 'ten', 'afterwards', 'sometime', 'whither', 'un', 'cant', 'hereupon', 'detail', 'must', 'about', 'bottom', 'while', 'eg', 'these', 'eight', 'due', 'five', 'never', 'we', 'nor', 'seemed', 'almost', 'over', 'put', 'ltd', 'ours', 'part', 'mine', 'your', 'twenty', 'those', 'neither', 'themselves', 'three', 'enough', 'always', 'him', 'below', 'became', 'amoungst', 'might', 'keep', 'beforehand', 'whether', 'he', 'interest', 'such', 'one', 'myself', 'to', 'latterly', 'however', 'mostly', 'co', 'become', 'thick', 'my', 'own', 'seeming', 'nothing', 'move', 'and', 'then', 'six', 'cry', 'moreover', 'sincere', 'wherein', 'again', 'together', 'been', 'within', 'done', 'further', 'that', 'indeed', 'next', 'thin', 'will', 'should', 'down', 'than', 'anyway', 'thence', 'in', 'any', 'once', 'another', 'often', 'up', 'before', 'from', 'go', 'thus', 'inc', 'describe', 'first', 'somehow', 'whereby', 'hereafter', 'seems', 'across', 'when', 'whereupon', 'forty', 'thereby', 'find', 'therefore', 'out', 'wherever', 'after', 'becomes', 'the', 'was', 'see', 'becoming', 'etc', 'perhaps', 'whoever', 'everywhere', 'though', 'through', 'hers', 'although', 'too', 'show', 'noone', 'but', 'until', 'all', 'whatever', 'with', 'without', 'otherwise', 'me', 'they', 'would', 'empty', 'being', 'four', 'under', 'she', 'herein', 'whenever', 'its', 'per', 'amongst', 'whole', 'against', 'her', 'everyone', 'twelve', 'since', 'eleven', 'also', 're', 'toward', 'every', 'both', 'himself', 'am', 'full', 'de', 'less', 'some', 'last', 'very', 'beside', 'between', 'an', 'give', 'had', 'off', 'a', 'no', 'few', 'name', 'much', 'meanwhile', 'well', 'along', 'how', 'alone', 'is', 'cannot', 'something', 'serious', 'still', 'system', 'several', 'whose', 'yours', 'else', 'could', 'besides', 'amount', 'made', 'sixty', 'beyond', 'namely', 'be', 'back', 'front', 'our', 'during', 'thereafter', 'behind', 'found', 'least', 'who', 'hundred', 'others', 'at', 'upon', 'can', 'somewhere', 'i', 'same', 'third', 'why', 'are', 'ever', 'whereafter', 'ourselves', 'two', 'latter', 'his', 'them', 'into', 'what', 'more', 'everything', 'whereas', 'so', 'formerly', 'were', 'either', 'hence', 'con', 'thru', 'among', 'top', 'rather', 'sometimes', 'anywhere', 'except', 'you', 'it', 'fill', 'because', 'itself', 'nine', 'towards', 'whence', 'bill', 'around', 'not', 'this', 'or', 'hereby', 'their', 'for', 'even', 'mill', 'couldnt', 'throughout', 'only', 'where', 'many', 'yet', 'nevertheless', 'whom', 'there', 'ie', 'on', 'us', 'has', 'of', 'seem', 'hasnt', 'elsewhere', 'please', 'have', 'if', 'former', 'do', 'therein', 'thereupon', 'take', 'other', 'fire', 'as'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'toward', 'their', 'three', 'themselves', 'etc', 'cannot', 'out', 'seem', 'about', 'therefore', 'thereupon', 'onto', 'afterwards', 'they', 'for', 'eight', 'others', 'it', 'to', 'myself', 'noone', 'each', 'when', 'empty', 'becoming', 'is', 'nowhere', 'himself', 'with', 'even', 'least', 'whoever', 'however', 'of', 'other', 'been', 'ever', 'well', 'further', 'ourselves', 'less', 'should', 'into', 'mill', 'amount', 'thus', 'by', 'yet', 'move', 'go', 'might', 'too', 'else', 'not', 'several', 'because', 'due', 'find', 'nevertheless', 'thereby', 'thru', 'beyond', 'perhaps', 'herein', 'herself', 'whence', 'co', 'found', 'done', 'otherwise', 'among', 'in', 'whereby', 'never', 'seeming', 'me', 'cant', 'somehow', 'six', 'sometimes', 'all', 'bill', 'them', 'much', 'four', 'whereupon', 'yourselves', 'yourself', 'must', 'still', 'may', 'made', 'bottom', 'another', 'sixty', 'everyone', 'whither', 'thence', 'hereby', 'except', 'serious', 'next', 'also', 'within', 'throughout', 'during', 'its', 'forty', 'cry', 'see', 'last', 'many', 'will', 'keep', 'where', 'mostly', 'ie', 'was', 'eleven', 'more', 'name', 'him', 'system', 'detail', 'how', 'indeed', 'hereafter', 'being', 'do', 'can', 'the', 'meanwhile', 'give', 'few', 'moreover', 'every', 'her', 'get', 'beside', 'no', 'full', 'we', 'ours', 'after', 'already', 'these', 'amoungst', 're', 'between', 'once', 'have', 'had', 'above', 'everywhere', 'enough', 'amongst', 'at', 'per', 'or', 'whereafter', 'anything', 'than', 'before', 'whatever', 'first', 'become', 'front', 'via', 'be', 'from', 'nor', 'ltd', 'any', 'back', 'formerly', 'anyhow', 'towards', 'below', 'inc', 'behind', 'wherever', 'your', 'thick', 'mine', 'an', 'would', 'twenty', 'though', 'almost', 'neither', 'former', 'con', 'both', 'along', 'sometime', 'whereas', 'under', 'he', 'but', 'up', 'take', 'wherein', 'seemed', 'around', 'whole', 'sincere', 'although', 'became', 'nothing', 'put', 'which', 'therein', 'alone', 'thereafter', 'could', 'since', 'so', 'on', 'call', 'nine', 'own', 'most', 'against', 'some', 'none', 'ten', 'un', 'one', 'us', 'has', 'describe', 'nobody', 'hence', 'without', 'everything', 'hereupon', 'two', 'a', 'together', 'am', 'same', 'very', 'anywhere', 'only', 'latterly', 'while', 'latter', 'this', 'anyone', 'such', 'were', 'down', 'becomes', 'through', 'off', 'my', 'our', 'anyway', 'someone', 'often', 'always', 'i', 'who', 'thin', 'and', 'those', 'somewhere', 'de', 'across', 'eg', 'whether', 'either', 'part', 'besides', 'yours', 'something', 'here', 'you', 'whose', 'couldnt', 'whom', 'fifteen', 'show', 'itself', 'rather', 'twelve', 'interest', 'she', 'hasnt', 'why', 'seems', 'upon', 'third', 'if', 'again', 'please', 'fill', 'there', 'side', 'hundred', 'fifty', 'top', 'namely', 'over', 'hers', 'until', 'his', 'beforehand', 'whenever', 'as', 'elsewhere', 'then', 'what', 'that', 'now', 'fire', 'are', 'five'}) instead.\n\n--------------------------------------------------------------------------------\n51838 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'re', 'therein', 'please', 'another', 'become', 'within', 'due', 'can', 'then', 'top', 'somehow', 'ltd', 'un', 'moreover', 'back', 'or', 'eleven', 'else', 'less', 'thick', 'everywhere', 'enough', 'must', 'these', 'thereafter', 'otherwise', 'part', 'beyond', 'whence', 'if', 'each', 'together', 'anyone', 'fire', 'beforehand', 'show', 'sometimes', 'mine', 'for', 'up', 'something', 'herself', 'eight', 'will', 'amongst', 'five', 'twenty', 'herein', 'under', 'several', 'which', 'anything', 'thin', 'six', 'seem', 'ourselves', 'while', 'somewhere', 'still', 'had', 'hereby', 'below', 'though', 'thus', 'hereafter', 'very', 'nothing', 'into', 'but', 'against', 'con', 'where', 'never', 'he', 'when', 'you', 'along', 'why', 'such', 'hence', 'formerly', 'seems', 'me', 'all', 'before', 'many', 'find', 'full', 'bottom', 'noone', 'amount', 'about', 'them', 'who', 'done', 'keep', 'thru', 'be', 'some', 'move', 'myself', 'only', 'even', 'already', 'hasnt', 'so', 'third', 'go', 'in', 'whatever', 'above', 'see', 'few', 'nine', 'sometime', 'name', 'how', 'latterly', 'was', 'from', 'it', 'during', 'that', 'its', 'besides', 'yours', 'ie', 'etc', 'ten', 'has', 'there', 'via', 'mill', 'and', 'own', 'call', 'being', 'everyone', 'last', 'itself', 'thereby', 'are', 'describe', 'made', 'my', 'what', 'twelve', 'her', 'whereupon', 'anyway', 'behind', 'at', 'towards', 'throughout', 'became', 'not', 'none', 'nor', 'whenever', 'latter', 'ever', 'bill', 'anyhow', 'us', 'hereupon', 'nowhere', 'same', 'the', 'becoming', 'i', 'sixty', 'seemed', 'a', 'empty', 'except', 'whether', 'whole', 'by', 'amoungst', 'most', 'get', 'indeed', 'put', 'any', 'out', 'therefore', 'around', 'since', 'sincere', 'as', 'after', 'two', 'himself', 'whereas', 'couldnt', 'side', 'is', 'over', 'although', 'former', 'almost', 'mostly', 'been', 'yet', 'nobody', 'further', 'well', 'might', 'also', 'meanwhile', 'whoever', 'could', 'alone', 'their', 'until', 'elsewhere', 'on', 'found', 'his', 'do', 'we', 'him', 'forty', 'without', 'of', 'yourselves', 'whereby', 'per', 'now', 'hers', 'de', 'fill', 'our', 'an', 'themselves', 'to', 'upon', 'someone', 'should', 'first', 'hundred', 'co', 'afterwards', 'whereafter', 'detail', 'down', 'whom', 'once', 'wherein', 'cant', 'whither', 'cry', 'namely', 'least', 'others', 'eg', 'seeming', 'than', 'nevertheless', 'other', 'she', 'through', 'those', 'have', 'were', 'give', 'fifteen', 'because', 'every', 'am', 'cannot', 'always', 'becomes', 'off', 'system', 'next', 'perhaps', 'no', 'rather', 'anywhere', 'more', 'front', 'this', 'again', 'inc', 'onto', 'whose', 'between', 'much', 'interest', 'fifty', 'neither', 'would', 'one', 'either', 'both', 'across', 'thereupon', 'serious', 'often', 'three', 'thence', 'however', 'toward', 'everything', 'beside', 'wherever', 'here', 'four', 'your', 'they', 'too', 'may', 'yourself', 'ours', 'with', 'among', 'take'}) instead.\n",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m/users/nfs/Etu7/21200397/Documents/M1 DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb Cell 9\u001b[0m line \u001b[0;36m3\n\u001b[1;32m     <a href='vscode-notebook-cell:/users/nfs/Etu7/21200397/Documents/M1%20DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb#X11sZmlsZQ%3D%3D?line=25'>26</a>\u001b[0m [X_all_train, X_all_test, Y_train, y_test]  \u001b[39m=\u001b[39m train_test_split(alltxts, alllabs, test_size\u001b[39m=\u001b[39m\u001b[39m0.2\u001b[39m, random_state\u001b[39m=\u001b[39m\u001b[39m10\u001b[39m, shuffle\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m)\n\u001b[1;32m     <a href='vscode-notebook-cell:/users/nfs/Etu7/21200397/Documents/M1%20DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb#X11sZmlsZQ%3D%3D?line=27'>28</a>\u001b[0m grid_search \u001b[39m=\u001b[39m GridSearchCV(pipeline, parameters,scoring\u001b[39m=\u001b[39mscoring, refit\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mf1_score\u001b[39m\u001b[39m'\u001b[39m, cv\u001b[39m=\u001b[39m\u001b[39m5\u001b[39m, n_jobs\u001b[39m=\u001b[39m\u001b[39m-\u001b[39m\u001b[39m1\u001b[39m, verbose\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[0;32m---> <a href='vscode-notebook-cell:/users/nfs/Etu7/21200397/Documents/M1%20DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb#X11sZmlsZQ%3D%3D?line=30'>31</a>\u001b[0m grid_search\u001b[39m.\u001b[39;49mfit(X_all_train, Y_train)\n\u001b[1;32m     <a href='vscode-notebook-cell:/users/nfs/Etu7/21200397/Documents/M1%20DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb#X11sZmlsZQ%3D%3D?line=33'>34</a>\u001b[0m \u001b[39mprint\u001b[39m(\u001b[39m\"\u001b[39m\u001b[39mMeilleurs paramètres trouvés:\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m     <a href='vscode-notebook-cell:/users/nfs/Etu7/21200397/Documents/M1%20DAC/RITAL/Projet/TAL-analyse-de-sentiments-et-reconnaissance-du-locuteur/Analyse_De_Sentiments_Movies/4_Grid_Search.ipynb#X11sZmlsZQ%3D%3D?line=34'>35</a>\u001b[0m \u001b[39mprint\u001b[39m(grid_search\u001b[39m.\u001b[39mbest_params_)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/dist-packages/sklearn/base.py:1351\u001b[0m, in \u001b[0;36m_fit_context.<locals>.decorator.<locals>.wrapper\u001b[0;34m(estimator, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1344\u001b[0m     estimator\u001b[39m.\u001b[39m_validate_params()\n\u001b[1;32m   1346\u001b[0m \u001b[39mwith\u001b[39;00m config_context(\n\u001b[1;32m   1347\u001b[0m     skip_parameter_validation\u001b[39m=\u001b[39m(\n\u001b[1;32m   1348\u001b[0m         prefer_skip_nested_validation \u001b[39mor\u001b[39;00m global_skip_validation\n\u001b[1;32m   1349\u001b[0m     )\n\u001b[1;32m   1350\u001b[0m ):\n\u001b[0;32m-> 1351\u001b[0m     \u001b[39mreturn\u001b[39;00m fit_method(estimator, \u001b[39m*\u001b[39;49margs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_search.py:970\u001b[0m, in \u001b[0;36mBaseSearchCV.fit\u001b[0;34m(self, X, y, **params)\u001b[0m\n\u001b[1;32m    964\u001b[0m     results \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_format_results(\n\u001b[1;32m    965\u001b[0m         all_candidate_params, n_splits, all_out, all_more_results\n\u001b[1;32m    966\u001b[0m     )\n\u001b[1;32m    968\u001b[0m     \u001b[39mreturn\u001b[39;00m results\n\u001b[0;32m--> 970\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_run_search(evaluate_candidates)\n\u001b[1;32m    972\u001b[0m \u001b[39m# multimetric is determined here because in the case of a callable\u001b[39;00m\n\u001b[1;32m    973\u001b[0m \u001b[39m# self.scoring the return type is only known after calling\u001b[39;00m\n\u001b[1;32m    974\u001b[0m first_test_score \u001b[39m=\u001b[39m all_out[\u001b[39m0\u001b[39m][\u001b[39m\"\u001b[39m\u001b[39mtest_scores\u001b[39m\u001b[39m\"\u001b[39m]\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_search.py:1527\u001b[0m, in \u001b[0;36mGridSearchCV._run_search\u001b[0;34m(self, evaluate_candidates)\u001b[0m\n\u001b[1;32m   1525\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_run_search\u001b[39m(\u001b[39mself\u001b[39m, evaluate_candidates):\n\u001b[1;32m   1526\u001b[0m \u001b[39m    \u001b[39m\u001b[39m\"\"\"Search all candidates in param_grid\"\"\"\u001b[39;00m\n\u001b[0;32m-> 1527\u001b[0m     evaluate_candidates(ParameterGrid(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mparam_grid))\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_search.py:947\u001b[0m, in \u001b[0;36mBaseSearchCV.fit.<locals>.evaluate_candidates\u001b[0;34m(candidate_params, cv, more_results)\u001b[0m\n\u001b[1;32m    940\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39mlen\u001b[39m(out) \u001b[39m!=\u001b[39m n_candidates \u001b[39m*\u001b[39m n_splits:\n\u001b[1;32m    941\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m    942\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mcv.split and cv.get_n_splits returned \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    943\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39minconsistent results. Expected \u001b[39m\u001b[39m{}\u001b[39;00m\u001b[39m \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    944\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39msplits, got \u001b[39m\u001b[39m{}\u001b[39;00m\u001b[39m\"\u001b[39m\u001b[39m.\u001b[39mformat(n_splits, \u001b[39mlen\u001b[39m(out) \u001b[39m/\u001b[39m\u001b[39m/\u001b[39m n_candidates)\n\u001b[1;32m    945\u001b[0m     )\n\u001b[0;32m--> 947\u001b[0m _warn_or_raise_about_fit_failures(out, \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49merror_score)\n\u001b[1;32m    949\u001b[0m \u001b[39m# For callable self.scoring, the return type is only know after\u001b[39;00m\n\u001b[1;32m    950\u001b[0m \u001b[39m# calling. If the return type is a dictionary, the error scores\u001b[39;00m\n\u001b[1;32m    951\u001b[0m \u001b[39m# can now be inserted with the correct key. The type checking\u001b[39;00m\n\u001b[1;32m    952\u001b[0m \u001b[39m# of out will be done in `_insert_error_scores`.\u001b[39;00m\n\u001b[1;32m    953\u001b[0m \u001b[39mif\u001b[39;00m callable(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mscoring):\n",
      "File \u001b[0;32m/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py:531\u001b[0m, in \u001b[0;36m_warn_or_raise_about_fit_failures\u001b[0;34m(results, error_score)\u001b[0m\n\u001b[1;32m    524\u001b[0m \u001b[39mif\u001b[39;00m num_failed_fits \u001b[39m==\u001b[39m num_fits:\n\u001b[1;32m    525\u001b[0m     all_fits_failed_message \u001b[39m=\u001b[39m (\n\u001b[1;32m    526\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39mAll the \u001b[39m\u001b[39m{\u001b[39;00mnum_fits\u001b[39m}\u001b[39;00m\u001b[39m fits failed.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    527\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mIt is very likely that your model is misconfigured.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    528\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mYou can try to debug the error by setting error_score=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mraise\u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\\n\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    529\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mBelow are more details about the failures:\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m{\u001b[39;00mfit_errors_summary\u001b[39m}\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    530\u001b[0m     )\n\u001b[0;32m--> 531\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(all_fits_failed_message)\n\u001b[1;32m    533\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    534\u001b[0m     some_fits_failed_message \u001b[39m=\u001b[39m (\n\u001b[1;32m    535\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m{\u001b[39;00mnum_failed_fits\u001b[39m}\u001b[39;00m\u001b[39m fits failed out of a total of \u001b[39m\u001b[39m{\u001b[39;00mnum_fits\u001b[39m}\u001b[39;00m\u001b[39m.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    536\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mThe score on these train-test partitions for these parameters\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    540\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mBelow are more details about the failures:\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m{\u001b[39;00mfit_errors_summary\u001b[39m}\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    541\u001b[0m     )\n",
      "\u001b[0;31mValueError\u001b[0m: \nAll the 829440 fits failed.\nIt is very likely that your model is misconfigured.\nYou can try to debug the error by setting error_score='raise'.\n\nBelow are more details about the failures:\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'sincere', 'ten', 'can', 'former', 'even', 'again', 'along', 'seemed', 'forty', 'no', 'seeming', 'whence', 'why', 'nobody', 'about', 'part', 'none', 'towards', 'whereafter', 'never', 'empty', 'therein', 'so', 'i', 'whither', 'cannot', 'however', 'itself', 'because', 'being', 'became', 'in', 'ourselves', 'ie', 'they', 'would', 'five', 'and', 'anyone', 'thus', 'around', 'still', 'sixty', 'now', 'indeed', 'done', 'six', 'will', 'sometime', 'per', 'beyond', 'among', 'such', 'if', 'less', 'moreover', 'behind', 'here', 'un', 'etc', 'an', 'yourselves', 'has', 'we', 'found', 'wherein', 'eg', 'amoungst', 'of', 'are', 'themselves', 'another', 'off', 'under', 'where', 'have', 'enough', 'who', 'seem', 'might', 'over', 'own', 'this', 'therefore', 'often', 'system', 'eleven', 'next', 'several', 're', 'above', 'himself', 'nowhere', 'yourself', 'hereafter', 'name', 'below', 'fire', 'out', 'anyhow', 'what', 'up', 'were', 'then', 'amongst', 'put', 'hence', 'after', 'from', 'few', 'her', 'do', 'formerly', 'made', 'most', 'myself', 'during', 'into', 'within', 'must', 'throughout', 'interest', 'everywhere', 'only', 'perhaps', 'be', 'else', 'between', 'least', 'also', 'four', 'nine', 'hereby', 'should', 'via', 'side', 'same', 'whose', 'one', 'she', 'other', 'but', 'not', 'thereby', 'con', 'when', 'latter', 'get', 'all', 'for', 'twenty', 'describe', 'somewhere', 'alone', 'my', 'neither', 'amount', 'yours', 'by', 'he', 'more', 'anyway', 'others', 'any', 'him', 'find', 'everyone', 'once', 'bill', 'thin', 'detail', 'them', 'herein', 'its', 'except', 'mine', 'latterly', 'whole', 'whatever', 'hundred', 'front', 'someone', 'fill', 'had', 'keep', 'us', 'through', 'everything', 'their', 'herself', 'give', 'eight', 'toward', 'anything', 'ever', 'each', 'wherever', 'fifteen', 'somehow', 'upon', 'though', 'whereupon', 'bottom', 'although', 'de', 'me', 'third', 'yet', 'every', 'rather', 'hereupon', 'your', 'fifty', 'otherwise', 'whom', 'whereby', 'thereafter', 'whether', 'against', 'go', 'as', 'both', 'some', 'at', 'how', 'nevertheless', 'ltd', 'afterwards', 'our', 'sometimes', 'cant', 'thereupon', 'may', 'without', 'is', 'see', 'almost', 'either', 'beforehand', 'full', 'that', 'mostly', 'to', 'these', 'nothing', 'until', 'couldnt', 'becoming', 'show', 'anywhere', 'meanwhile', 'before', 'the', 'a', 'hasnt', 'or', 'was', 'those', 'last', 'whoever', 'elsewhere', 'thence', 'across', 'very', 'twelve', 'been', 'first', 'always', 'namely', 'seems', 'become', 'cry', 'two', 'move', 'there', 'with', 'since', 'serious', 'am', 'whenever', 'on', 'due', 'becomes', 'mill', 'noone', 'back', 'it', 'nor', 'too', 'three', 'which', 'please', 'while', 'onto', 'whereas', 'besides', 'thru', 'call', 'something', 'than', 'inc', 'already', 'his', 'hers', 'ours', 'co', 'thick', 'take', 'down', 'many', 'together', 'top', 'could', 'you', 'further', 'well', 'much', 'beside'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'anything', 'over', 'since', 'whom', 'neither', 'hers', 'whereas', 'anyhow', 'whereby', 'front', 'herself', 'our', 'due', 'next', 'amount', 'after', 'describe', 'hundred', 'rather', 'none', 'a', 'am', 'it', 'hence', 'nobody', 'eleven', 'along', 'mine', 'anyone', 'therefore', 'across', 'together', 'de', 'toward', 'until', 'on', 'then', 'again', 'fifteen', 'less', 'become', 'thereafter', 'your', 'onto', 'most', 'yet', 'eight', 'what', 'may', 'without', 'becomes', 'mill', 'was', 'find', 'namely', 'ever', 'up', 'seems', 'wherein', 'four', 'afterwards', 'those', 'detail', 'cry', 'thin', 'bill', 'cant', 'fire', 'whence', 'fifty', 'more', 'go', 'under', 'put', 'us', 'ten', 'themselves', 'bottom', 'besides', 'seemed', 'top', 'noone', 'even', 'my', 'do', 'much', 'empty', 'part', 'enough', 'everything', 'seem', 'only', 'moreover', 'towards', 'its', 'name', 'sometime', 'above', 'their', 'every', 'eg', 'around', 'ltd', 'though', 'through', 'this', 'beside', 'are', 'three', 'least', 'somewhere', 'wherever', 'sometimes', 'therein', 'con', 'me', 'during', 'and', 'another', 'him', 'co', 'full', 'he', 'we', 'made', 'who', 'his', 'sixty', 'whose', 'has', 'thick', 'among', 'ourselves', 'six', 'both', 'yourselves', 'either', 'once', 'although', 'see', 'somehow', 'other', 're', 'everywhere', 'all', 'from', 'hereby', 'hasnt', 'is', 'be', 'any', 'move', 'ie', 'why', 'system', 'amoungst', 'whoever', 'one', 'whole', 'anyway', 'i', 'often', 'been', 'within', 'fill', 'because', 'etc', 'take', 'yours', 'which', 'below', 'hereupon', 'latterly', 'whereafter', 'she', 'whenever', 'could', 'thereby', 'by', 'two', 'first', 'get', 'out', 'last', 'already', 'had', 'serious', 'thus', 'give', 'here', 'beforehand', 'about', 'alone', 'where', 'as', 'behind', 'keep', 'forty', 'anywhere', 'twelve', 'mostly', 'how', 'however', 'amongst', 'former', 'nevertheless', 'never', 'the', 'interest', 'so', 'down', 'but', 'sincere', 'yourself', 'also', 'into', 'show', 'further', 'herein', 'via', 'thru', 'formerly', 'seeming', 'no', 'cannot', 'will', 'with', 'there', 'against', 'these', 'couldnt', 'otherwise', 'something', 'her', 'except', 'should', 'must', 'whatever', 'of', 'have', 'an', 'each', 'many', 'such', 'found', 'almost', 'third', 'whereupon', 'beyond', 'five', 'became', 'well', 'others', 'hereafter', 'nor', 'were', 'when', 'might', 'they', 'or', 'myself', 'always', 'per', 'off', 'everyone', 'inc', 'own', 'please', 'done', 'too', 'now', 'thereupon', 'in', 'very', 'you', 'for', 'than', 'twenty', 'at', 'if', 'that', 'latter', 'someone', 'some', 'whether', 'to', 'them', 'nothing', 'himself', 'still', 'upon', 'else', 'between', 'itself', 'ours', 'elsewhere', 'call', 'throughout', 'nowhere', 'being', 'while', 'becoming', 'un', 'same', 'thence', 'indeed', 'whither', 'would', 'before', 'perhaps', 'not', 'several', 'side', 'meanwhile', 'can', 'nine', 'few', 'back'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'along', 'full', 'most', 'this', 'was', 'in', 'because', 'while', 'forty', 're', 'on', 'if', 'yourselves', 'must', 'find', 'against', 'itself', 'those', 'cant', 'by', 'thru', 'ten', 'back', 'be', 'fill', 'former', 'due', 'but', 'himself', 'elsewhere', 'three', 'ie', 'it', 'otherwise', 'everything', 'amount', 'however', 'mill', 'has', 'anyone', 'off', 'therefore', 'always', 'about', 'herself', 'wherein', 'above', 'somehow', 'why', 'detail', 'inc', 'who', 'hasnt', 'done', 'have', 'twelve', 'four', 'sometime', 'ever', 'seems', 'which', 'would', 'thereupon', 'latter', 'can', 'becoming', 'ourselves', 'take', 'towards', 'whereby', 'whereas', 'interest', 'your', 'more', 'were', 'whenever', 'nor', 'become', 'own', 'already', 'down', 'you', 'hence', 'show', 'them', 'describe', 'whereafter', 'being', 'until', 'ltd', 'beforehand', 'whole', 'his', 'me', 'first', 'all', 'their', 'something', 'myself', 'or', 'anywhere', 'via', 'namely', 'might', 'these', 'any', 'thence', 'never', 'together', 'once', 'within', 'couldnt', 'latterly', 'no', 'de', 'someone', 'are', 'only', 'amongst', 'thin', 'few', 'nine', 'among', 'every', 'around', 'fire', 'give', 'nobody', 'mostly', 'twenty', 'get', 'five', 'moreover', 'themselves', 'so', 'whereupon', 'eleven', 'per', 'enough', 'see', 'toward', 'same', 'that', 'how', 'its', 'for', 'seem', 'sincere', 'across', 'behind', 'put', 'nevertheless', 'had', 'out', 'whether', 'seeming', 'could', 'nothing', 'onto', 'top', 'indeed', 'nowhere', 'both', 'none', 'with', 'i', 'one', 'third', 'whom', 'bill', 'well', 'some', 'found', 'another', 'before', 'side', 'afterwards', 'then', 'amoungst', 'wherever', 'eight', 'a', 'con', 'such', 'everyone', 'not', 'they', 'anything', 'keep', 'without', 'hereafter', 'therein', 'noone', 'herein', 'least', 'empty', 'often', 'through', 'fifty', 'beyond', 'is', 'thereafter', 'besides', 'hers', 'whither', 'here', 'at', 'into', 'and', 'etc', 'perhaps', 'am', 'as', 'move', 'much', 'been', 'each', 'anyway', 'her', 'name', 'many', 'upon', 'below', 'everywhere', 'too', 'fifteen', 'may', 'although', 'of', 'less', 'what', 'whatever', 'to', 'whence', 'now', 'sometimes', 'should', 'neither', 'hereby', 'throughout', 'my', 'between', 'when', 'anyhow', 'very', 'cannot', 'thick', 'do', 'hereupon', 'two', 'us', 'last', 'next', 'we', 'formerly', 'serious', 'meanwhile', 'go', 'under', 'than', 'yet', 'up', 'system', 'over', 'will', 'after', 'except', 'whose', 'sixty', 'further', 'from', 'she', 'six', 'thereby', 'during', 'alone', 'somewhere', 'several', 'beside', 'since', 'yourself', 'whoever', 'front', 'co', 'seemed', 'an', 'the', 'rather', 'made', 'even', 'hundred', 'bottom', 'mine', 'almost', 'eg', 'again', 'else', 'our', 'un', 'also', 'part', 'where', 'please', 'others', 'cry', 'either', 'still', 'thus', 'call', 'though', 'becomes', 'him', 'ours', 'there', 'yours', 'he', 'other', 'became'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'done', 'hereby', 'eg', 'front', 'found', 'someone', 'amongst', 'nor', 'until', 'ltd', 'within', 'among', 'so', 'other', 'do', 'either', 'can', 'cannot', 'amount', 'thereupon', 'whenever', 'during', 'herein', 'twelve', 'many', 'onto', 'may', 'whole', 'my', 'them', 'were', 'its', 'four', 'therefore', 'what', 'latterly', 'with', 'hers', 'go', 'if', 'beyond', 'not', 'nevertheless', 'themselves', 'this', 'thence', 'co', 'one', 'once', 'none', 'why', 'above', 'could', 'however', 'mostly', 'very', 'although', 'formerly', 'thereby', 'ourselves', 'hundred', 'when', 'serious', 'beside', 'who', 'myself', 'such', 'your', 'becomes', 'thus', 'therein', 'first', 'nowhere', 'off', 'yours', 'take', 'how', 'next', 'had', 'me', 'five', 'that', 'whereupon', 'call', 'seem', 'ever', 'amoungst', 'i', 'couldnt', 'some', 'than', 'further', 'all', 'any', 'only', 'detail', 'the', 'sixty', 'describe', 'full', 'inc', 'whereafter', 'well', 'indeed', 'becoming', 'perhaps', 'meanwhile', 'everyone', 'am', 'please', 'most', 'de', 'etc', 'which', 'thru', 'throughout', 'no', 'she', 'on', 'yourselves', 'against', 'seemed', 'seems', 'towards', 'another', 'already', 'whence', 'cry', 'it', 'been', 'own', 'have', 'name', 'elsewhere', 'whatever', 'out', 'mill', 'get', 'thick', 'mine', 'often', 'sincere', 'thereafter', 'yet', 'twenty', 'enough', 'in', 'ours', 'third', 'except', 'part', 'her', 'up', 'afterwards', 'neither', 'under', 'latter', 'give', 'again', 'was', 'every', 'less', 'put', 'being', 'noone', 'back', 'then', 'fifty', 'hereupon', 'sometime', 'we', 'anyway', 'into', 'around', 'together', 'they', 'fill', 'everything', 'namely', 'whether', 'interest', 'behind', 'also', 'whereas', 'beforehand', 'through', 'wherein', 'always', 'almost', 'even', 'because', 'forty', 'whose', 'ie', 'us', 'per', 'side', 'last', 'fire', 'before', 'will', 'see', 'moreover', 'be', 'else', 'bottom', 'below', 'become', 'as', 'more', 'fifteen', 'both', 'him', 'though', 'hasnt', 'others', 'due', 'whereby', 'somewhere', 're', 'here', 'least', 'nobody', 'con', 'few', 'to', 'over', 'system', 'of', 'three', 'still', 'whoever', 'became', 'must', 'besides', 'you', 'top', 'itself', 'made', 'while', 'two', 'former', 'bill', 'toward', 'eleven', 'too', 'between', 'much', 'an', 'for', 'something', 'and', 'from', 'cant', 'eight', 'empty', 'our', 'move', 'about', 'there', 'his', 'are', 'thin', 'hereafter', 'whither', 'sometimes', 'otherwise', 'un', 'along', 'seeming', 'alone', 'same', 'down', 'anyone', 'yourself', 'keep', 'find', 'upon', 'these', 'without', 'would', 'across', 'anything', 'or', 'by', 'anyhow', 'several', 'is', 'wherever', 'where', 'those', 'herself', 'hence', 'somehow', 'each', 'but', 'after', 'ten', 'he', 'should', 'everywhere', 'himself', 'since', 'via', 'has', 'might', 'never', 'a', 'now', 'nine', 'whom', 'six', 'rather', 'show', 'at', 'nothing', 'their', 'anywhere'}) instead.\n\n--------------------------------------------------------------------------------\n51842 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'when', 'indeed', 'something', 'here', 'someone', 'do', 'eleven', 'become', 'across', 'i', 'too', 'except', 'such', 'up', 'himself', 'show', 'several', 'done', 'empty', 'in', 'although', 'anyhow', 'whence', 'after', 'there', 'behind', 'other', 'under', 'an', 'ie', 'see', 'hers', 'describe', 'ours', 'by', 'within', 'few', 'hereafter', 'less', 'detail', 'any', 'top', 'each', 'without', 'where', 'can', 'hence', 'me', 'his', 'herein', 'serious', 'un', 'meanwhile', 'many', 'sometimes', 'besides', 'you', 'amongst', 'yourself', 'last', 'along', 'why', 'enough', 'around', 'part', 'whether', 'then', 'for', 'against', 'together', 'everything', 'been', 'are', 'one', 'find', 'everyone', 'more', 'their', 'be', 'whatever', 'same', 'keep', 'else', 'until', 'latterly', 'thereafter', 'bill', 'nor', 'she', 'thin', 'nobody', 'back', 'could', 'due', 'yours', 'everywhere', 'your', 'ltd', 'sixty', 'to', 'almost', 'between', 'three', 'full', 'ourselves', 'while', 'should', 'as', 'via', 'anything', 'must', 'has', 'him', 'because', 'moreover', 'itself', 'now', 'six', 'con', 'whereupon', 'amount', 'fire', 'nothing', 'give', 'thus', 'inc', 'eight', 'nine', 'becomes', 'former', 'how', 'forty', 'among', 'is', 'very', 'side', 're', 'found', 'might', 'if', 'afterwards', 'he', 'seemed', 'of', 'thence', 'who', 'cry', 'either', 'beside', 'go', 'was', 'perhaps', 'anyway', 'would', 'co', 'through', 'with', 'so', 'towards', 'next', 'however', 'sometime', 'twelve', 'name', 'themselves', 'seems', 'eg', 'a', 'down', 'than', 'thick', 'anyone', 'every', 'never', 'but', 'therein', 'those', 'cant', 'others', 'since', 'much', 'wherever', 'seem', 'neither', 'all', 'further', 'above', 'get', 'first', 'made', 'take', 'formerly', 'interest', 'we', 'even', 'once', 'thru', 'below', 'whenever', 'whom', 'de', 'nevertheless', 'least', 'twenty', 'will', 'third', 'move', 'always', 'otherwise', 'beyond', 'fifteen', 'five', 'only', 'elsewhere', 'had', 'another', 'put', 'what', 'hasnt', 'alone', 'became', 'anywhere', 'mill', 'own', 'upon', 'throughout', 'already', 'per', 'ever', 'well', 'most', 'off', 'whose', 'yourselves', 'am', 'none', 'whereafter', 'they', 'again', 'herself', 'still', 'this', 'have', 'therefore', 'my', 'thereby', 'toward', 'our', 'namely', 'becoming', 'at', 'wherein', 'us', 'mine', 'it', 'couldnt', 'ten', 'whoever', 'though', 'seeming', 'two', 'etc', 'rather', 'noone', 'nowhere', 'hereby', 'whereas', 'bottom', 'also', 'which', 'often', 'please', 'latter', 'being', 'fill', 'somewhere', 'over', 'about', 'system', 'some', 'cannot', 'fifty', 'myself', 'thereupon', 'not', 'sincere', 'beforehand', 'onto', 'that', 'somehow', 'no', 'on', 'hereupon', 'them', 'may', 'amoungst', 'into', 'her', 'whole', 'its', 'mostly', 'during', 'these', 'whereby', 'call', 'both', 'out', 'were', 'and', 'before', 'front', 'four', 'yet', 'or', 'whither', 'from', 'hundred', 'the'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'is', 'ie', 'your', 'should', 'since', 'call', 'be', 'eg', 'when', 'most', 'us', 'two', 'against', 'own', 'whither', 'after', 'enough', 'first', 'elsewhere', 'about', 'other', 'detail', 'together', 'put', 'who', 'thin', 'always', 'either', 'on', 'rather', 'you', 'meanwhile', 'whence', 'thence', 'except', 'because', 'whereupon', 'between', 'seeming', 'whoever', 'cannot', 'last', 'mill', 'fire', 'herself', 'whereas', 'few', 'through', 'became', 'along', 'with', 'we', 'moreover', 'being', 'hereafter', 'therein', 'or', 'empty', 'please', 'whose', 'too', 'not', 'had', 'could', 'hence', 'among', 'around', 'amongst', 'and', 'twelve', 'themselves', 'as', 'sincere', 'he', 'of', 'take', 'whenever', 'my', 'never', 'neither', 'get', 'its', 'go', 'have', 'couldnt', 'one', 'would', 'full', 'before', 'whole', 'am', 'seems', 'no', 'across', 'where', 'upon', 'down', 'but', 'yourself', 'something', 'an', 'bottom', 'each', 'although', 'thereafter', 'wherein', 'for', 'fifty', 'latter', 'inc', 'therefore', 'etc', 'whatever', 'ours', 'else', 'hers', 'three', 'sometimes', 'they', 'formerly', 'former', 'it', 'whom', 'up', 'becomes', 'myself', 'however', 'alone', 'this', 'system', 'over', 'find', 'per', 'his', 'interest', 'the', 'amount', 'though', 'due', 'found', 'him', 'forty', 'con', 'off', 'wherever', 'anyone', 'both', 'hundred', 'give', 'besides', 'yourselves', 'more', 'another', 'everyone', 'toward', 'perhaps', 'many', 'here', 'nobody', 'were', 'some', 'again', 'hereupon', 'can', 'name', 'even', 'becoming', 'must', 'throughout', 'fill', 'has', 'ever', 'everywhere', 'nevertheless', 'indeed', 'itself', 'if', 'only', 'ten', 'via', 'whereby', 'mostly', 'further', 'whereafter', 'until', 're', 'why', 'amoungst', 'serious', 'also', 'others', 'well', 'anyhow', 'will', 'that', 'de', 'above', 'mine', 'during', 'into', 'beside', 'yours', 'nor', 'i', 'nine', 'see', 'four', 'hasnt', 'out', 'whether', 'while', 'any', 'already', 'do', 'ourselves', 'are', 'how', 'namely', 'very', 'show', 'eleven', 'latterly', 'thereupon', 'them', 'same', 'which', 'seem', 'beforehand', 'thru', 'move', 'somehow', 'next', 'within', 'keep', 'least', 'third', 'below', 'describe', 'almost', 'bill', 'beyond', 'those', 'eight', 'all', 'onto', 'yet', 'then', 'noone', 'sixty', 'by', 'part', 'now', 'still', 'done', 'top', 'cant', 'himself', 'towards', 'everything', 'back', 'she', 'somewhere', 'ltd', 'than', 'in', 'what', 'me', 'might', 'six', 'thick', 'anywhere', 'twenty', 'none', 'such', 'often', 'thereby', 'anything', 'co', 'so', 'nowhere', 'herein', 'un', 'every', 'there', 'sometime', 'thus', 'under', 'a', 'hereby', 'otherwise', 'less', 'seemed', 'our', 'may', 'made', 'five', 'these', 'nothing', 'her', 'side', 'several', 'cry', 'once', 'become', 'their', 'front', 'behind', 'at', 'from', 'without', 'anyway', 'someone', 'afterwards', 'to', 'was', 'much', 'been', 'fifteen'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'should', 'each', 'eight', 'four', 'wherein', 'cant', 'are', 'to', 'describe', 'whole', 'thereupon', 'keep', 'been', 'indeed', 'moreover', 'ever', 'mill', 'by', 'with', 'go', 'de', 'become', 'anyone', 'latterly', 'these', 'though', 'hence', 'thin', 'among', 'wherever', 'some', 'because', 'every', 'the', 'either', 'therefore', 'whereas', 'un', 'done', 'beyond', 'before', 'sincere', 'otherwise', 'noone', 'thereafter', 'i', 'onto', 'twenty', 'was', 'himself', 'would', 'often', 'mostly', 'inc', 'ours', 'nor', 'must', 'has', 'nine', 'latter', 'me', 'amount', 'yours', 'below', 'find', 'myself', 'few', 'over', 'former', 'both', 'afterwards', 'someone', 'under', 'found', 'ltd', 'itself', 'bottom', 'of', 'please', 'fill', 'for', 'without', 'thence', 'as', 'my', 'cry', 'something', 'yet', 'whenever', 'into', 'anyway', 'here', 'mine', 'am', 'after', 'nobody', 'everyone', 'meanwhile', 'eg', 'within', 'very', 'system', 'what', 'five', 'fifty', 'therein', 'or', 'their', 'one', 'whereafter', 'her', 'up', 'name', 'always', 'even', 'yourself', 'nothing', 'have', 'across', 'eleven', 'others', 'made', 'etc', 'last', 'becoming', 'more', 'can', 'too', 'amoungst', 'alone', 'then', 'empty', 'in', 'against', 'whatever', 'throughout', 'sixty', 'us', 'everything', 'six', 'namely', 'had', 'his', 'side', 'since', 'own', 'except', 'sometimes', 'seeming', 'out', 'and', 'via', 'might', 'how', 'our', 'do', 'move', 'together', 'where', 'is', 'anything', 'thereby', 'further', 'those', 'perhaps', 'seems', 'another', 'they', 'he', 'many', 'see', 'least', 'who', 'through', 'due', 'while', 'cannot', 'however', 'anyhow', 'almost', 'sometime', 'somehow', 'formerly', 'whoever', 'fire', 'a', 'still', 'amongst', 'between', 'towards', 'whereupon', 'much', 'your', 'which', 'else', 'thus', 'forty', 'hereupon', 'thru', 'whereby', 'besides', 'them', 'most', 'you', 'above', 'at', 'couldnt', 'none', 'themselves', 'its', 'top', 'along', 'again', 'off', 'part', 'there', 'about', 'once', 'show', 'she', 'if', 'never', 'detail', 'when', 'seemed', 'any', 'upon', 'back', 'it', 'rather', 're', 'full', 'two', 'hereby', 'hers', 'whether', 'herein', 'than', 'behind', 'so', 'well', 'an', 'now', 'somewhere', 'why', 'became', 'next', 'elsewhere', 'neither', 'yourselves', 'call', 'him', 'also', 'from', 'twelve', 'we', 'being', 'ten', 'be', 'other', 'no', 'take', 'whence', 'anywhere', 'three', 'nevertheless', 'get', 'hundred', 'ie', 'bill', 'fifteen', 'becomes', 'whom', 'this', 'co', 'several', 'third', 'were', 'per', 'whose', 'beside', 'during', 'less', 'around', 'will', 'only', 'serious', 'on', 'put', 'interest', 'thick', 'whither', 'nowhere', 'down', 'could', 'hasnt', 'same', 'until', 'not', 'give', 'that', 'con', 'such', 'all', 'although', 'herself', 'everywhere', 'ourselves', 'first', 'but', 'toward', 'may', 'seem', 'enough', 'already', 'front', 'hereafter', 'beforehand'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'four', 'de', 'seems', 'they', 'out', 'fifty', 'be', 'first', 'empty', 'are', 'became', 'himself', 'describe', 'although', 'nobody', 'something', 'him', 'been', 'were', 'that', 'thereupon', 'only', 'meanwhile', 'formerly', 'moreover', 'yourself', 'but', 'well', 'the', 'perhaps', 'somewhere', 'forty', 'inc', 'into', 'more', 'cant', 'hence', 'than', 'have', 'everything', 'in', 'them', 're', 'serious', 'one', 'it', 'thereby', 'anything', 'ltd', 'keep', 'ie', 'eight', 'top', 'otherwise', 'if', 'call', 'where', 'hers', 'i', 'eg', 'though', 'find', 'yourselves', 'next', 'may', 'alone', 'third', 'your', 'indeed', 'whereupon', 'cry', 'to', 'six', 'should', 'hereby', 'etc', 'whole', 'whose', 'off', 'as', 'hereafter', 'side', 'and', 'beforehand', 'give', 'now', 'become', 'least', 'co', 'until', 'full', 'together', 'others', 'mine', 'am', 'which', 'themselves', 'myself', 'sometime', 'whereby', 'do', 'nine', 'sincere', 'ever', 'latterly', 'wherever', 'no', 'last', 'whenever', 'seeming', 'for', 'an', 'this', 'mill', 'every', 'within', 'many', 'has', 'down', 'onto', 'ten', 'namely', 'seem', 'on', 'everywhere', 'however', 'who', 'go', 'anyhow', 'fifteen', 'whatever', 'whereas', 'twelve', 'via', 'beside', 'mostly', 'bill', 'amount', 'towards', 'us', 'besides', 'both', 'during', 'whither', 'most', 'upon', 'not', 'rather', 'how', 'since', 'what', 'enough', 'or', 'across', 'often', 'system', 'take', 'several', 'couldnt', 'show', 'too', 'hereupon', 'throughout', 'nowhere', 'by', 'around', 'three', 'interest', 'seemed', 'would', 'our', 'found', 'per', 'his', 'hasnt', 'from', 'wherein', 'somehow', 'at', 'between', 'never', 'becoming', 'was', 'still', 'her', 'someone', 'under', 'even', 'before', 'had', 'thick', 'through', 'about', 'again', 'such', 'two', 'noone', 'none', 'ourselves', 'almost', 'nor', 'yours', 'she', 'afterwards', 'own', 'anywhere', 'anyone', 'toward', 'why', 'against', 'amongst', 'move', 'very', 'along', 'whence', 'is', 'being', 'any', 'already', 'hundred', 'up', 'there', 'its', 'among', 'other', 'much', 'please', 'might', 'my', 'everyone', 'amoungst', 'yet', 'another', 'he', 'nothing', 'those', 'over', 'see', 'then', 'when', 'their', 'beyond', 'these', 'always', 'less', 'also', 'so', 'behind', 'cannot', 'will', 'thin', 'sometimes', 'therefore', 'un', 'thus', 'due', 'name', 'fire', 'whereafter', 'con', 'else', 'detail', 'below', 'sixty', 'few', 'latter', 'eleven', 'you', 'ours', 'five', 'back', 'we', 'herein', 'herself', 'while', 'further', 'without', 'all', 'of', 'after', 'because', 'twenty', 'except', 'done', 'here', 'whether', 'must', 'elsewhere', 'once', 'either', 'some', 'with', 'can', 'made', 'itself', 'above', 'fill', 'neither', 'whom', 'could', 'former', 'put', 'therein', 'thereafter', 'becomes', 'get', 'anyway', 'same', 'thru', 'me', 'part', 'each', 'nevertheless', 'thence', 'front', 'a', 'bottom', 'whoever'}) instead.\n\n--------------------------------------------------------------------------------\n51840 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'cant', 'much', 'every', 'first', 'several', 'had', 'how', 'hers', 'side', 'around', 'eg', 'keep', 'onto', 'fifteen', 'either', 'few', 'mine', 'here', 'now', 'during', 'were', 'etc', 'enough', 'thereafter', 'you', 'these', 'between', 'ever', 'whereby', 'me', 'while', 'beside', 'she', 'hereupon', 'everyone', 'must', 'indeed', 'most', 'behind', 'give', 'four', 'very', 'though', 'before', 'alone', 'with', 'bill', 'been', 'out', 'although', 'detail', 'being', 'too', 'becomes', 'this', 'only', 'his', 'where', 'hereby', 'describe', 'by', 'do', 'it', 'myself', 'eleven', 'nothing', 'fill', 'more', 'whereupon', 'perhaps', 'each', 'another', 'would', 'to', 'since', 'all', 'whatever', 'ltd', 'the', 'always', 'towards', 'never', 'and', 'thus', 're', 'co', 'whenever', 'eight', 'somewhere', 'then', 'should', 'everything', 'something', 'whither', 'we', 'have', 'without', 'who', 'hereafter', 'was', 'no', 'also', 'may', 'because', 'take', 'nevertheless', 'meanwhile', 'twelve', 'down', 'against', 'done', 'when', 'he', 'herein', 'through', 'whom', 'anyhow', 'get', 'twenty', 'into', 'latterly', 'fifty', 'yours', 'bottom', 'my', 'full', 'interest', 'still', 'elsewhere', 'found', 'could', 'within', 'hasnt', 'nor', 'whoever', 'might', 'amoungst', 'seems', 'forty', 'due', 'until', 'among', 'six', 'for', 'about', 'moreover', 'next', 'over', 'which', 'inc', 'rather', 'per', 'less', 'throughout', 'below', 'a', 'almost', 'others', 'back', 'her', 'see', 'however', 'else', 'why', 'name', 'nine', 'move', 'is', 'us', 'neither', 'cannot', 'un', 'yourselves', 'third', 'seemed', 'along', 'thick', 'whose', 'whether', 'namely', 'sometime', 'has', 'what', 'mill', 'already', 'be', 'are', 'any', 'fire', 'at', 'itself', 'even', 'if', 'but', 'thereby', 'such', 'off', 'once', 'toward', 'via', 'one', 'so', 'wherein', 'an', 'except', 'serious', 'beyond', 'sincere', 'afterwards', 'ten', 'go', 'top', 'hundred', 'nobody', 'wherever', 'him', 'am', 'front', 'latter', 'anywhere', 'across', 'will', 'part', 'none', 'thru', 'whence', 'some', 'yourself', 'or', 'yet', 'somehow', 'again', 'empty', 'under', 'became', 'they', 'thence', 'de', 'thin', 'call', 'system', 'them', 'own', 'upon', 'amongst', 'its', 'anyone', 'many', 'as', 'otherwise', 'cry', 'that', 'other', 'former', 'therefore', 'everywhere', 'there', 'herself', 'put', 'those', 'our', 'than', 'two', 'show', 'after', 'of', 'anyway', 'three', 'made', 'often', 'formerly', 'whereafter', 'thereupon', 'i', 'ie', 'seem', 'noone', 'themselves', 'becoming', 'please', 'your', 'their', 'from', 'anything', 'find', 'ours', 'besides', 'amount', 'whole', 'on', 'least', 'five', 'both', 'whereas', 'same', 'ourselves', 'hence', 'in', 'further', 'seeming', 'someone', 'mostly', 'beforehand', 'well', 'couldnt', 'together', 'up', 'can', 'sometimes', 'con', 'sixty', 'last', 'himself', 'become', 'not', 'therein', 'nowhere', 'above'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'along', 'his', 'same', 'whoever', 'anyone', 'former', 'anything', 'whom', 'everything', 'put', 'via', 'among', 'if', 'these', 'ours', 'will', 'an', 'due', 'into', 'well', 'herein', 'we', 'five', 'am', 'nevertheless', 'anyway', 'formerly', 'cant', 'how', 'such', 'between', 'my', 'amount', 'nowhere', 'very', 'is', 'after', 'side', 'whereafter', 'always', 'themselves', 'neither', 'bill', 'should', 'a', 'by', 'there', 'hereupon', 'hundred', 'the', 'all', 'except', 'whole', 'besides', 'many', 'whose', 'few', 'has', 'us', 'every', 'seemed', 'often', 'of', 'your', 'sixty', 'anyhow', 'who', 'amoungst', 'please', 'are', 'when', 'upon', 'ten', 'must', 'somewhere', 'thereafter', 'whether', 'sometime', 'however', 'de', 'nobody', 'both', 'three', 'out', 'find', 'before', 'fill', 'un', 'becomes', 'whatever', 'whence', 'four', 'whither', 'most', 'eg', 'but', 're', 'part', 'or', 'throughout', 'nothing', 'get', 'hereafter', 'something', 'latter', 'namely', 'beside', 'co', 'also', 'other', 'inc', 'below', 'found', 'meanwhile', 'serious', 'give', 'so', 'third', 'cry', 'keep', 'can', 'more', 'our', 'about', 'afterwards', 'him', 'whereas', 'been', 'may', 'not', 'go', 'whereupon', 'twelve', 'whereby', 'con', 'off', 'in', 'moreover', 'further', 'seems', 'be', 'nor', 'fire', 'detail', 'and', 'their', 'onto', 'back', 'last', 'see', 'done', 'because', 'otherwise', 'whenever', 'wherever', 'he', 'me', 'through', 'move', 'have', 'only', 'no', 'hers', 'which', 'describe', 'etc', 'too', 'indeed', 'own', 'rather', 'it', 'thru', 'she', 'to', 'even', 'everywhere', 'sometimes', 'down', 'as', 'this', 'across', 'myself', 'somehow', 'ltd', 'you', 'bottom', 'becoming', 'being', 'front', 'under', 'none', 'yourself', 'eight', 'above', 'amongst', 'her', 'behind', 'forty', 'others', 'seeming', 'either', 'fifteen', 'again', 'some', 'per', 'they', 'therefore', 'couldnt', 'nine', 'why', 'someone', 'else', 'show', 'them', 'cannot', 'twenty', 'here', 'might', 'then', 'interest', 'another', 'while', 'with', 'those', 'beyond', 'that', 'its', 'from', 'beforehand', 'each', 'up', 'herself', 'thence', 'six', 'though', 'noone', 'was', 'latterly', 'never', 'one', 'itself', 'i', 'elsewhere', 'take', 'eleven', 'first', 'for', 'next', 'within', 'everyone', 'thereupon', 'mostly', 'where', 'system', 'already', 'could', 'therein', 'together', 'two', 'than', 'wherein', 'would', 'fifty', 'hence', 'any', 'seem', 'yours', 'mine', 'himself', 'become', 'ourselves', 'until', 'do', 'alone', 'over', 'almost', 'now', 'full', 'still', 'at', 'during', 'had', 'mill', 'much', 'least', 'thick', 'toward', 'thereby', 'once', 'without', 'were', 'ever', 'yet', 'perhaps', 'on', 'anywhere', 'made', 'since', 'sincere', 'ie', 'less', 'towards', 'hasnt', 'although', 'several', 'name', 'thin', 'empty', 'enough', 'top', 'hereby', 'call', 'thus', 'yourselves', 'became', 'around', 'against', 'what'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'were', 'in', 'sometime', 'therefore', 'be', 'his', 'less', 'thereupon', 'otherwise', 'might', 'fire', 'yet', 'top', 'somehow', 'next', 'at', 'all', 'i', 'put', 'too', 'by', 'bottom', 'beforehand', 'hasnt', 'could', 'please', 'nevertheless', 'became', 'throughout', 'mine', 'even', 'fifteen', 'wherein', 'should', 'would', 'anyone', 'made', 'me', 'behind', 'thence', 'thereafter', 'along', 'ours', 'may', 'hers', 'formerly', 'yourselves', 'thick', 'not', 'whom', 'something', 'four', 'yourself', 'than', 'thus', 'among', 'eg', 'below', 'neither', 'with', 'every', 'others', 'myself', 'already', 'least', 'you', 'has', 'him', 'off', 'back', 'move', 'name', 'almost', 'again', 'and', 'therein', 'seeming', 'moreover', 'everything', 'is', 'such', 'enough', 'other', 'forty', 'perhaps', 'often', 'get', 'herself', 'my', 'some', 'noone', 'done', 'us', 'sometimes', 'do', 'the', 'everywhere', 'seemed', 'themselves', 'whereas', 'after', 'then', 'etc', 'amount', 'six', 'either', 'somewhere', 'except', 'twelve', 'onto', 'against', 're', 'are', 'same', 'once', 'serious', 'amongst', 'through', 'fill', 'third', 'sixty', 'ie', 'so', 'upon', 'beyond', 'hence', 'due', 'nor', 'go', 'was', 'becomes', 'mostly', 'which', 'amoungst', 'being', 'if', 'becoming', 'a', 'further', 'ourselves', 'must', 'several', 'anyway', 'well', 'as', 'that', 'also', 'why', 'yours', 'nothing', 'couldnt', 'into', 'someone', 'detail', 'whatever', 'give', 'your', 'meanwhile', 'since', 'five', 'will', 'she', 'but', 'together', 'eight', 'now', 'no', 'few', 'beside', 'over', 'toward', 'what', 'he', 'here', 'besides', 'during', 'itself', 'whereupon', 'ten', 'out', 'another', 'very', 'become', 'afterwards', 'alone', 'everyone', 'still', 'cant', 'rather', 'on', 'inc', 'side', 'this', 'twenty', 'part', 'three', 'up', 'none', 'himself', 'ever', 'can', 'more', 'show', 'interest', 'across', 'around', 'been', 'whose', 'towards', 'anything', 'without', 'mill', 'whither', 'for', 'see', 'nobody', 'who', 'we', 'from', 'call', 'where', 'nine', 'when', 'of', 'former', 'last', 'herein', 'it', 'to', 'hereby', 'else', 'anywhere', 'keep', 'thin', 'whence', 'before', 'take', 'empty', 'one', 'down', 'front', 'only', 'any', 'eleven', 'they', 'thru', 'ltd', 'indeed', 'many', 'its', 'elsewhere', 'above', 'hundred', 'anyhow', 'via', 'our', 'these', 'bill', 'system', 'about', 'find', 'nowhere', 'wherever', 'much', 'cry', 'hereafter', 'never', 'per', 'co', 'con', 'though', 'their', 'namely', 'thereby', 'an', 'while', 'under', 'latterly', 'how', 'most', 'them', 'seems', 'had', 'between', 'however', 'whoever', 'whereby', 'de', 'those', 'whereafter', 'am', 'describe', 'whether', 'whenever', 'first', 'cannot', 'or', 'found', 'seem', 'own', 'although', 'fifty', 'whole', 'both', 'each', 'sincere', 'always', 'latter', 'un', 'there', 'because', 'until', 'her', 'full', 'two', 'hereupon', 'have', 'within'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'found', 'ours', 'everything', 'alone', 'around', 'perhaps', 'same', 'ever', 'nor', 'else', 'latterly', 'seemed', 'all', 'eg', 'either', 'whole', 'side', 'her', 'whereafter', 'up', 'everywhere', 'some', 'was', 'along', 'as', 'more', 'therefore', 'fifteen', 'thereafter', 'otherwise', 'former', 'each', 'latter', 'sixty', 'two', 'thick', 'other', 'whom', 'nothing', 'never', 'top', 'why', 'has', 'go', 'me', 'his', 'move', 'too', 'mill', 'into', 'next', 'others', 'had', 'were', 'became', 'no', 'above', 'you', 'moreover', 'twelve', 'both', 'thereupon', 'further', 'ten', 'before', 'since', 'whoever', 'very', 'therein', 'itself', 'thence', 'about', 'could', 'so', 'nevertheless', 'un', 'behind', 'thereby', 'interest', 'in', 'becomes', 'a', 'beforehand', 'much', 'forty', 'enough', 'between', 'everyone', 'also', 'they', 'might', 'none', 'one', 'them', 'wherein', 'thus', 'full', 'hereby', 'us', 'yourselves', 'con', 'part', 'hereafter', 'become', 'couldnt', 'there', 'somewhere', 'their', 'how', 'eight', 'if', 'across', 'when', 'bottom', 'formerly', 'fill', 'hasnt', 'few', 'towards', 'via', 'themselves', 'off', 'bill', 'always', 'until', 'although', 'three', 'last', 'seeming', 'something', 'hence', 'made', 'besides', 'herself', 'yet', 'least', 'ourselves', 'de', 'namely', 'however', 'under', 'anyway', 'most', 'show', 'whither', 'its', 'even', 'am', 'that', 'another', 'from', 'must', 'my', 'toward', 'but', 'beyond', 'put', 'hers', 'here', 'she', 'over', 'below', 'co', 'ltd', 'though', 'during', 'again', 'once', 'on', 'amongst', 'per', 'thru', 'these', 'sometime', 'third', 'amoungst', 'within', 'together', 'seem', 'i', 'ie', 'seems', 'the', 'whose', 'yourself', 'anywhere', 'upon', 'already', 'whence', 'have', 'hereupon', 'name', 'five', 'call', 'detail', 'is', 'throughout', 'because', 'somehow', 'well', 'give', 'describe', 'he', 'every', 'down', 'it', 'still', 'inc', 'whereby', 'mostly', 'anyone', 'whatever', 'beside', 'may', 'front', 'after', 'who', 'four', 'eleven', 'nowhere', 'many', 'etc', 'wherever', 'get', 'and', 'afterwards', 'himself', 'to', 'only', 'less', 'what', 'cry', 'thin', 'for', 'now', 'whether', 'against', 'at', 'him', 'system', 'out', 'do', 'back', 'almost', 'herein', 'indeed', 'not', 'due', 'this', 'someone', 'nine', 'being', 'neither', 'please', 'by', 'fire', 'elsewhere', 'your', 'rather', 'myself', 'nobody', 'find', 'than', 'where', 'take', 'we', 'are', 'empty', 'six', 'will', 'whereupon', 'should', 'cannot', 'anyhow', 'first', 'except', 'twenty', 'serious', 'sincere', 'done', 'which', 'with', 'several', 'keep', 'onto', 'been', 'cant', 'without', 'mine', 'be', 'whereas', 'noone', 'through', 'yours', 'our', 'hundred', 'amount', 'among', 'own', 'can', 'becoming', 'would', 'anything', 'of', 'an', 're', 'meanwhile', 'such', 'see', 'then', 'whenever', 'often', 'while', 'fifty', 'sometimes', 'or', 'those', 'any'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'thick', 'my', 'become', 'same', 'below', 'about', 'sometimes', 'describe', 'amount', 'made', 'own', 'less', 'others', 'behind', 'sixty', 'etc', 'nowhere', 'a', 'within', 'yours', 'ltd', 'so', 'keep', 'ours', 'they', 'becoming', 'their', 'whom', 'his', 'whereafter', 'the', 'too', 'anywhere', 'during', 'either', 'formerly', 'but', 'what', 'ie', 'she', 'anyway', 'one', 'himself', 'against', 'four', 'move', 'i', 'was', 'seems', 'could', 'cannot', 'otherwise', 'find', 'whereupon', 'get', 'somewhere', 'again', 'most', 'whereas', 'mostly', 'un', 'are', 'already', 'thence', 'since', 'neither', 'herself', 'nor', 'two', 'them', 'together', 'wherever', 'alone', 'everywhere', 'than', 'whatever', 'her', 'be', 'per', 'inc', 'someone', 'some', 'thus', 'therefore', 'call', 'whenever', 'eight', 'namely', 'also', 'if', 'front', 'with', 'over', 'whence', 'once', 'whoever', 'along', 'hereupon', 'next', 'though', 'elsewhere', 'may', 'nothing', 'de', 'toward', 'else', 'twenty', 'hereafter', 'yourselves', 'around', 'those', 'although', 're', 'whether', 'latter', 'empty', 'co', 'would', 'besides', 'twelve', 'well', 'of', 'detail', 'without', 'out', 'mine', 'under', 'fill', 'why', 'put', 'always', 'we', 'where', 'he', 'never', 'full', 'hereby', 'something', 'more', 'indeed', 'while', 'that', 'myself', 'eg', 'name', 'somehow', 'found', 'latterly', 'amoungst', 'cry', 'will', 'back', 'several', 'might', 'must', 'thru', 'has', 'three', 'towards', 'amongst', 'or', 'couldnt', 'see', 'am', 'many', 'third', 'further', 'before', 'top', 'much', 'interest', 'hence', 'hers', 'among', 'him', 'should', 'seeming', 'then', 'few', 'another', 'very', 'whose', 'as', 'do', 'via', 'this', 'none', 'at', 'eleven', 'across', 'until', 'nobody', 'above', 'yet', 'six', 'thereupon', 'last', 'nevertheless', 'least', 'whereby', 'these', 'any', 'afterwards', 'up', 'it', 'except', 'and', 'hundred', 'often', 'anyone', 'an', 'mill', 'ourselves', 'former', 'when', 'side', 'seemed', 'sincere', 'show', 'onto', 'whither', 'to', 'not', 'by', 'seem', 'after', 'itself', 'beside', 'fifty', 'throughout', 'thin', 'everyone', 'have', 'yourself', 'noone', 'you', 'how', 'even', 'only', 'its', 'done', 'each', 'however', 'here', 'us', 'now', 'take', 'go', 'becomes', 'down', 'which', 'fire', 'is', 'on', 'almost', 'anything', 'hasnt', 'there', 'in', 'fifteen', 'give', 'who', 'both', 'first', 'sometime', 'themselves', 'wherein', 'con', 'because', 'bottom', 'no', 'anyhow', 'being', 'nine', 'due', 'ten', 'into', 'part', 'such', 'me', 'forty', 'our', 'perhaps', 'every', 'upon', 'all', 'ever', 'been', 'herein', 'therein', 'can', 'beyond', 'became', 'everything', 'bill', 'please', 'serious', 'beforehand', 'from', 'system', 'through', 'meanwhile', 'were', 'your', 'thereafter', 'enough', 'other', 'still', 'had', 'whole', 'thereby', 'between', 'cant', 'off', 'five', 'rather', 'for', 'moreover'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'via', 'already', 'anyone', 'above', 'most', 'may', 'nobody', 'onto', 'yourself', 'none', 'fifteen', 'nowhere', 'call', 'now', 'yourselves', 'by', 'fifty', 'here', 'each', 'herself', 'which', 'get', 'anyhow', 'someone', 'anything', 'side', 'ten', 'afterwards', 'sometime', 'whither', 'un', 'cant', 'hereupon', 'detail', 'must', 'about', 'bottom', 'while', 'eg', 'these', 'eight', 'due', 'five', 'never', 'we', 'nor', 'seemed', 'almost', 'over', 'put', 'ltd', 'ours', 'part', 'mine', 'your', 'twenty', 'those', 'neither', 'themselves', 'three', 'enough', 'always', 'him', 'below', 'became', 'amoungst', 'might', 'keep', 'beforehand', 'whether', 'he', 'interest', 'such', 'one', 'myself', 'to', 'latterly', 'however', 'mostly', 'co', 'become', 'thick', 'my', 'own', 'seeming', 'nothing', 'move', 'and', 'then', 'six', 'cry', 'moreover', 'sincere', 'wherein', 'again', 'together', 'been', 'within', 'done', 'further', 'that', 'indeed', 'next', 'thin', 'will', 'should', 'down', 'than', 'anyway', 'thence', 'in', 'any', 'once', 'another', 'often', 'up', 'before', 'from', 'go', 'thus', 'inc', 'describe', 'first', 'somehow', 'whereby', 'hereafter', 'seems', 'across', 'when', 'whereupon', 'forty', 'thereby', 'find', 'therefore', 'out', 'wherever', 'after', 'becomes', 'the', 'was', 'see', 'becoming', 'etc', 'perhaps', 'whoever', 'everywhere', 'though', 'through', 'hers', 'although', 'too', 'show', 'noone', 'but', 'until', 'all', 'whatever', 'with', 'without', 'otherwise', 'me', 'they', 'would', 'empty', 'being', 'four', 'under', 'she', 'herein', 'whenever', 'its', 'per', 'amongst', 'whole', 'against', 'her', 'everyone', 'twelve', 'since', 'eleven', 'also', 're', 'toward', 'every', 'both', 'himself', 'am', 'full', 'de', 'less', 'some', 'last', 'very', 'beside', 'between', 'an', 'give', 'had', 'off', 'a', 'no', 'few', 'name', 'much', 'meanwhile', 'well', 'along', 'how', 'alone', 'is', 'cannot', 'something', 'serious', 'still', 'system', 'several', 'whose', 'yours', 'else', 'could', 'besides', 'amount', 'made', 'sixty', 'beyond', 'namely', 'be', 'back', 'front', 'our', 'during', 'thereafter', 'behind', 'found', 'least', 'who', 'hundred', 'others', 'at', 'upon', 'can', 'somewhere', 'i', 'same', 'third', 'why', 'are', 'ever', 'whereafter', 'ourselves', 'two', 'latter', 'his', 'them', 'into', 'what', 'more', 'everything', 'whereas', 'so', 'formerly', 'were', 'either', 'hence', 'con', 'thru', 'among', 'top', 'rather', 'sometimes', 'anywhere', 'except', 'you', 'it', 'fill', 'because', 'itself', 'nine', 'towards', 'whence', 'bill', 'around', 'not', 'this', 'or', 'hereby', 'their', 'for', 'even', 'mill', 'couldnt', 'throughout', 'only', 'where', 'many', 'yet', 'nevertheless', 'whom', 'there', 'ie', 'on', 'us', 'has', 'of', 'seem', 'hasnt', 'elsewhere', 'please', 'have', 'if', 'former', 'do', 'therein', 'thereupon', 'take', 'other', 'fire', 'as'}) instead.\n\n--------------------------------------------------------------------------------\n51839 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'toward', 'their', 'three', 'themselves', 'etc', 'cannot', 'out', 'seem', 'about', 'therefore', 'thereupon', 'onto', 'afterwards', 'they', 'for', 'eight', 'others', 'it', 'to', 'myself', 'noone', 'each', 'when', 'empty', 'becoming', 'is', 'nowhere', 'himself', 'with', 'even', 'least', 'whoever', 'however', 'of', 'other', 'been', 'ever', 'well', 'further', 'ourselves', 'less', 'should', 'into', 'mill', 'amount', 'thus', 'by', 'yet', 'move', 'go', 'might', 'too', 'else', 'not', 'several', 'because', 'due', 'find', 'nevertheless', 'thereby', 'thru', 'beyond', 'perhaps', 'herein', 'herself', 'whence', 'co', 'found', 'done', 'otherwise', 'among', 'in', 'whereby', 'never', 'seeming', 'me', 'cant', 'somehow', 'six', 'sometimes', 'all', 'bill', 'them', 'much', 'four', 'whereupon', 'yourselves', 'yourself', 'must', 'still', 'may', 'made', 'bottom', 'another', 'sixty', 'everyone', 'whither', 'thence', 'hereby', 'except', 'serious', 'next', 'also', 'within', 'throughout', 'during', 'its', 'forty', 'cry', 'see', 'last', 'many', 'will', 'keep', 'where', 'mostly', 'ie', 'was', 'eleven', 'more', 'name', 'him', 'system', 'detail', 'how', 'indeed', 'hereafter', 'being', 'do', 'can', 'the', 'meanwhile', 'give', 'few', 'moreover', 'every', 'her', 'get', 'beside', 'no', 'full', 'we', 'ours', 'after', 'already', 'these', 'amoungst', 're', 'between', 'once', 'have', 'had', 'above', 'everywhere', 'enough', 'amongst', 'at', 'per', 'or', 'whereafter', 'anything', 'than', 'before', 'whatever', 'first', 'become', 'front', 'via', 'be', 'from', 'nor', 'ltd', 'any', 'back', 'formerly', 'anyhow', 'towards', 'below', 'inc', 'behind', 'wherever', 'your', 'thick', 'mine', 'an', 'would', 'twenty', 'though', 'almost', 'neither', 'former', 'con', 'both', 'along', 'sometime', 'whereas', 'under', 'he', 'but', 'up', 'take', 'wherein', 'seemed', 'around', 'whole', 'sincere', 'although', 'became', 'nothing', 'put', 'which', 'therein', 'alone', 'thereafter', 'could', 'since', 'so', 'on', 'call', 'nine', 'own', 'most', 'against', 'some', 'none', 'ten', 'un', 'one', 'us', 'has', 'describe', 'nobody', 'hence', 'without', 'everything', 'hereupon', 'two', 'a', 'together', 'am', 'same', 'very', 'anywhere', 'only', 'latterly', 'while', 'latter', 'this', 'anyone', 'such', 'were', 'down', 'becomes', 'through', 'off', 'my', 'our', 'anyway', 'someone', 'often', 'always', 'i', 'who', 'thin', 'and', 'those', 'somewhere', 'de', 'across', 'eg', 'whether', 'either', 'part', 'besides', 'yours', 'something', 'here', 'you', 'whose', 'couldnt', 'whom', 'fifteen', 'show', 'itself', 'rather', 'twelve', 'interest', 'she', 'hasnt', 'why', 'seems', 'upon', 'third', 'if', 'again', 'please', 'fill', 'there', 'side', 'hundred', 'fifty', 'top', 'namely', 'over', 'hers', 'until', 'his', 'beforehand', 'whenever', 'as', 'elsewhere', 'then', 'what', 'that', 'now', 'fire', 'are', 'five'}) instead.\n\n--------------------------------------------------------------------------------\n51838 fits failed with the following error:\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/model_selection/_validation.py\", line 890, in _fit_and_score\n    estimator.fit(X_train, y_train, **fit_params)\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1351, in wrapper\n    return fit_method(estimator, *args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 471, in fit\n    Xt = self._fit(X, y, routed_params)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 408, in _fit\n    X, fitted_transformer = fit_transform_one_cached(\n                            ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/joblib/memory.py\", line 353, in __call__\n    return self.func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/pipeline.py\", line 1303, in _fit_transform_one\n    res = transformer.fit_transform(X, y, **params.get(\"fit_transform\", {}))\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/feature_extraction/text.py\", line 2138, in fit_transform\n    X = super().fit_transform(raw_documents)\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 1344, in wrapper\n    estimator._validate_params()\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/base.py\", line 666, in _validate_params\n    validate_parameter_constraints(\n  File \"/usr/local/lib/python3.11/dist-packages/sklearn/utils/_param_validation.py\", line 95, in validate_parameter_constraints\n    raise InvalidParameterError(\nsklearn.utils._param_validation.InvalidParameterError: The 'stop_words' parameter of TfidfVectorizer must be a str among {'english'}, an instance of 'list' or None. Got frozenset({'re', 'therein', 'please', 'another', 'become', 'within', 'due', 'can', 'then', 'top', 'somehow', 'ltd', 'un', 'moreover', 'back', 'or', 'eleven', 'else', 'less', 'thick', 'everywhere', 'enough', 'must', 'these', 'thereafter', 'otherwise', 'part', 'beyond', 'whence', 'if', 'each', 'together', 'anyone', 'fire', 'beforehand', 'show', 'sometimes', 'mine', 'for', 'up', 'something', 'herself', 'eight', 'will', 'amongst', 'five', 'twenty', 'herein', 'under', 'several', 'which', 'anything', 'thin', 'six', 'seem', 'ourselves', 'while', 'somewhere', 'still', 'had', 'hereby', 'below', 'though', 'thus', 'hereafter', 'very', 'nothing', 'into', 'but', 'against', 'con', 'where', 'never', 'he', 'when', 'you', 'along', 'why', 'such', 'hence', 'formerly', 'seems', 'me', 'all', 'before', 'many', 'find', 'full', 'bottom', 'noone', 'amount', 'about', 'them', 'who', 'done', 'keep', 'thru', 'be', 'some', 'move', 'myself', 'only', 'even', 'already', 'hasnt', 'so', 'third', 'go', 'in', 'whatever', 'above', 'see', 'few', 'nine', 'sometime', 'name', 'how', 'latterly', 'was', 'from', 'it', 'during', 'that', 'its', 'besides', 'yours', 'ie', 'etc', 'ten', 'has', 'there', 'via', 'mill', 'and', 'own', 'call', 'being', 'everyone', 'last', 'itself', 'thereby', 'are', 'describe', 'made', 'my', 'what', 'twelve', 'her', 'whereupon', 'anyway', 'behind', 'at', 'towards', 'throughout', 'became', 'not', 'none', 'nor', 'whenever', 'latter', 'ever', 'bill', 'anyhow', 'us', 'hereupon', 'nowhere', 'same', 'the', 'becoming', 'i', 'sixty', 'seemed', 'a', 'empty', 'except', 'whether', 'whole', 'by', 'amoungst', 'most', 'get', 'indeed', 'put', 'any', 'out', 'therefore', 'around', 'since', 'sincere', 'as', 'after', 'two', 'himself', 'whereas', 'couldnt', 'side', 'is', 'over', 'although', 'former', 'almost', 'mostly', 'been', 'yet', 'nobody', 'further', 'well', 'might', 'also', 'meanwhile', 'whoever', 'could', 'alone', 'their', 'until', 'elsewhere', 'on', 'found', 'his', 'do', 'we', 'him', 'forty', 'without', 'of', 'yourselves', 'whereby', 'per', 'now', 'hers', 'de', 'fill', 'our', 'an', 'themselves', 'to', 'upon', 'someone', 'should', 'first', 'hundred', 'co', 'afterwards', 'whereafter', 'detail', 'down', 'whom', 'once', 'wherein', 'cant', 'whither', 'cry', 'namely', 'least', 'others', 'eg', 'seeming', 'than', 'nevertheless', 'other', 'she', 'through', 'those', 'have', 'were', 'give', 'fifteen', 'because', 'every', 'am', 'cannot', 'always', 'becomes', 'off', 'system', 'next', 'perhaps', 'no', 'rather', 'anywhere', 'more', 'front', 'this', 'again', 'inc', 'onto', 'whose', 'between', 'much', 'interest', 'fifty', 'neither', 'would', 'one', 'either', 'both', 'across', 'thereupon', 'serious', 'often', 'three', 'thence', 'however', 'toward', 'everything', 'beside', 'wherever', 'here', 'four', 'your', 'they', 'too', 'may', 'yourself', 'ours', 'with', 'among', 'take'}) instead.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "pipeline = Pipeline([\n",
    "    ('tfidf', TfidfVectorizer(stop_words='english')),\n",
    "    ('reg', LogisticRegression())\n",
    "])\n",
    "\n",
    "parameters = {\n",
    "    'tfidf__max_df': [0.5, 0.6, 0.7, 0.8, 0.9, 1.0],\n",
    "    'tfidf__min_df': [2, 3, 5, 10],\n",
    "    'tfidf__ngram_range': [(1, 1), (1, 2), (1, 3), (2, 2), (2, 3) , (3,3)],\n",
    "    'tfidf__binary': [False, True],\n",
    "    'tfidf__lowercase': [False, True],\n",
    "    'tfidf__use_idf': [False, True],\n",
    "    'tfidf__sublinear_tf': [False, True],\n",
    "    'tfidf__max_features': [None, 2000, 5000, 10000, 20000 , 40000],\n",
    "\n",
    "    'reg__C': [0.1, 1, 10, 100],\n",
    "    'reg__penalty': ['none', 'l2' , 'l1']\n",
    "}\n",
    "\n",
    "scoring = {\n",
    "    'f1_score': make_scorer(f1_score),\n",
    "    'roc_auc': make_scorer(roc_auc_score),\n",
    "    'accuracy': make_scorer(accuracy_score)\n",
    "}\n",
    "\n",
    "[X_all_train, X_all_test, Y_train, y_test]  = train_test_split(alltxts, alllabs, test_size=0.2, random_state=10, shuffle=True)\n",
    "\n",
    "grid_search = GridSearchCV(pipeline, parameters,scoring=scoring, refit='f1_score', cv=5, n_jobs=-1, verbose=1)\n",
    "\n",
    "\n",
    "grid_search.fit(X_all_train, Y_train)\n",
    "\n",
    "\n",
    "print(\"Meilleurs paramètres trouvés:\")\n",
    "print(grid_search.best_params_)\n",
    "\n",
    "\n",
    "print(\"Scores:\")\n",
    "print(\"F1 Score:\", grid_search.cv_results_['mean_test_f1_score'])\n",
    "print(\"AUC:\", grid_search.cv_results_['mean_test_roc_auc'])\n",
    "print(\"Accuracy:\", grid_search.cv_results_['mean_test_accuracy'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Grid search pour la régression logistic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = lambda doc: ut.suppression_balises_html( ut.remove_numbers(ut.remove_ponctuation(doc)) ) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Meilleurs hyperparamètres trouvés :  {'C': 100, 'penalty': 'l2'}\n",
      "\n",
      "Scores moyens pour chaque métrique :\n",
      "f1_score: [       nan 0.33471933 0.83159619        nan 0.42948704 0.83159619\n",
      "        nan 0.78704815 0.83159619        nan 0.8192636  0.83159619\n",
      "        nan 0.83730742 0.83159619        nan 0.83981059 0.83159619]\n",
      "roc_auc: [       nan 0.8709559  0.90873862        nan 0.87145592 0.90873862\n",
      "        nan 0.87513575 0.90873862        nan 0.89444119 0.90873862\n",
      "        nan 0.91209032 0.90873862        nan 0.91563733 0.90873862]\n",
      "accuracy: [     nan 0.503125 0.831875      nan 0.546875 0.831875      nan 0.788125\n",
      " 0.831875      nan 0.819375 0.831875      nan 0.8375   0.831875      nan\n",
      " 0.84     0.831875]\n"
     ]
    }
   ],
   "source": [
    "lr_grid_params = {'penalty': ['l1','l2' , None],\n",
    "          'C': [0.001, 0.01, 0.1, 1, 10, 100]}\n",
    "\n",
    "tf_idf_vectorizer = TfidfVectorizer(preprocessor=f, ngram_range=(1, 3), stop_words='english',tokenizer=word_tokenize, lowercase=True, max_features=40000)\n",
    "[X_all_train, X_all_test, Y_train, y_test]  = train_test_split(alltxts, alllabs, test_size=0.2, random_state=10, shuffle=True)\n",
    "X_train = tf_idf_vectorizer.fit_transform(X_all_train)\n",
    "X_test = tf_idf_vectorizer.transform(X_all_test)\n",
    "\n",
    "lr_clf = LogisticRegression(max_iter=100)\n",
    "\n",
    "\n",
    "lr_grid_params = {'penalty': ['l1','l2' , None],\n",
    "          'C': [0.001, 0.01, 0.1, 1, 10, 100]}\n",
    "\n",
    "scoring = {\n",
    "    'f1_score': make_scorer(f1_score, average='macro'),  # F1 score\n",
    "    'roc_auc': make_scorer(roc_auc_score, needs_proba=True, multi_class='ovr'),  # AUC\n",
    "    'accuracy': make_scorer(accuracy_score)  # Accuracy\n",
    "}\n",
    "\n",
    "lr_grid_search = GridSearchCV(lr_clf, param_grid=lr_grid_params, cv=5 , scoring=scoring, refit='f1_score')\n",
    "\n",
    "lr_grid_search.fit(X_train, Y_train)\n",
    "\n",
    "print('Meilleurs hyperparamètres trouvés : ', lr_grid_search.best_params_)\n",
    "print()\n",
    "# Affichage des scores moyens pour chaque métrique\n",
    "results = lr_grid_search.cv_results_\n",
    "print(\"Scores moyens pour chaque métrique :\")\n",
    "for metric in ['f1_score', 'roc_auc', 'accuracy']:\n",
    "    print(f\"{metric}: {results[f'mean_test_{metric}']}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
